---
title: "Example workflow"
author: "Anna Quaglieri"
output:
  rmarkdown::html_vignette:
    fig_caption: yes
    toc_float: true
vignette: >
  %\VignetteIndexEntry{Vignette Title}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---


```{r setup, include=FALSE,eval=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


```{r eval=FALSE}
# indels 
library(tidyverse)

# Example data
# In the example below:
# I added a key for an indel including chr_pos_alt allele
# I added a variant ID just for this example to quickly inspect what variants should be left in 
# Below only variants A and B should be left in since C does not pass the filters in any sample
# The remission sample is never called by the caller


########################
## Create sample data
########################

# example indels
indels <- data.frame(PID = rep("P1",9),
                     Time = c(rep("Screen",3),rep("Cyc1",3),rep("Cyc2",3)),
                     Location = c("chr1_10","chr1_20","chr2_30","chr1_10","chr1_20","chr2_30","chr1_10","chr1_20","chr2_30"),
                     alt = c("ACT","ACGTCG","AGG","ACT","ACGTCG","AGG","ACT","ACGTCG","AGG"),
                     ref_depth = c(12,11,9,9,12,8,13,14,8),
                     vaf = rep(0.7,9)) %>% 
  mutate(variant_ID = c("A","B","C","A","B","C","A","B","C")) %>%
  unite(key, Location,alt,sep="_",remove=FALSE) 

# example clinical
clinical <- data.frame(PID = rep("P1",4),
                       Time = c("Screen","Rem","Cyc1","Cyc2"))

```



```{r}
library(varikondo)
example_data
```

