---
title: "Getting started with standardising and visualising your variants output"
author: "Anna Quaglieri"
date: "`r Sys.Date()`"
output: rmarkdown::html_vignette
bibliography: biblio.bib
fig_caption: yes
vignette: >
  %\VignetteIndexEntry{Vignette Title}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r setup, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  include = TRUE,
  echo=TRUE,
  comment = "#>"
)
```

# Background and purpuse

This package was developed by the necessity to standardise the output from several variant callers used to call variants in tumor-only RNA-Seq samples. In our research we focused on patients with multiple samples over time during treatment. The final aim was to have data in a suitable format to produce informative visualisation of the variant allele frequency (VAF) (or other features) of the mutations over time, here the name `lineplots`. There would be in principle no difference if variants were called using DNA-Seq. `lineplots` supports ouputs from the following callers run in tumour-only mode: `GATK3 MuTect2`, `VarScan2`, `VarDict` and `km`. It also supports slightly modified output from `superFreq`. Below is a graphic summary of the main purpose behind `lineplots`.

**add pic**

## A multidude of outputs

Focusing on RNA-Seq, calling variants from RNA-Seq samples is increasing in popularity due to the large number of RNA-Seq data produced but only exploited for gene expression **[papers]**. Several benchmarking papers have been published comparing the performance of different callers [] and in recent years several new methods have been developed specifically to improve INDEL detection in RNA-Seq samples []. As part of a project done on two Leukemia clinical trials we assembled a pipeline to call variants in RNA-Seq (https://github.com/annaquaglieri16/RNA-seq-variant-calling) and use [superFreq](https://github.com/ChristofferFlensburg/superFreq) to analyse clonal tracking and mutational development across treatment. We used combined calls from [VarDict](https://github.com/AstraZeneca-NGS/VarDict) [Lai2016-ws] and [km](https://github.com/iric-soft/km/tree/master/km/tools#output-description-1) [@Software-km] algorithm to define INDEL calls.  

After calling SNVs, CNVs and INDELs with different programs, it is often useful to combine all the results in a standardised way. Usually different programs output variants in different ways using slightly different conventions which makes it challenging to easily combine the results. For example, superFreq returns SNVs in `csv` files but CNVs in `Rdata` files; VarDict uses the standard `VCF` output but with different specification of quality measures compared to, for example, MuTect2 or VarScan2; km outputs INDELs in tab delimited format and so on. This makes it challenging to both create a comprehensive overview of the results for one sample as well as to compare results across callers. Another challenge comes from calling variants independently at different time points for the same patient. Variants might be present before treatment but absent after treatment and when a patient relapse or vice versa. We found it necessary to fill in all the missing time points especially for visualisations purposes. While superFreq takes this problem into account by keeping any variants found at different time points, other programs like km or VarDict donâ€™t. 
 
## Time course data

# Setup and available functions

The `lineplots` R package comprises several helper and import functions that can be used to convert output from any of the callers mentioned above to a standardised format. 

In general, the import functions take in input variants for one `patientID` and for a set of `studyGenes` called by a caller as well as clinical information for all the patients in the cohort under investigation. The ouput can be in a tidy format, setting `tidy = TRUE`, where each row corresponds to a mutation for one samples and columns are samples and variant informations. `studygenes` are expected to be NCBI gene `Symbols` which means that variants need to be annotated with the gene they belong to before `lineplots` can be used. The clinical data at every time point is needed because if, for a patient, a variant is not detected at one time point, then only clinical information will be returned with default values for the variant information. This is useful for vidualisation purposes where one needs to track the development of mutation over time.

```{r eval=FALSE,include=TRUE,echo=TRUE}
library(devtools)
devtools::install_github("annaquaglieri16/lineplots",force = TRUE)
```

```{r}
#detach("package:lineplots")
library(lineplots)
```


## Parsing functions: from `VCF` to tidy format

The function `parse_vcf_output()` allows to convert a `.vcf` file generated by three different callers (`GATK3 MuTect2`, `VarDict` and `Samtools + VarScan2`) to a data frame with standardised fields. This is because different callers annotate the `INFO` field in the `VCF` ouput in different ways, using different names for read depth, variant allele frequency etc... The output from `parse_vcf_output()` can then be used for `import_any_for_lineplots()`. If the variants in the `VCF` file were also annotated with the Variant Effect Predictor (VEP) [@McLaren2016-lv], the `parse_vep_csq()` can be subsequently used to parse the extra fields. 

It is to be noted that VEP usually annotate the same variant in multiple ways, depending on whether a variants falls on several transcripts. VEP pastes together different annotations on the same line. The `parse_vep_csq()` will reshape the data and return it in a long format by stacking all the annotations one underneath each other. This has the potential of largely increasing the size of the ouput file and it is suggested that, if the input `VCF` file is too large, to, for example, subset variants based on genes of interest.   

Both parsing functions are built on top of the [`VariantAnnotation`](https://bioconductor.org/packages/release/bioc/html/VariantAnnotation.html) Bioconductor package [@Obenchain2014-cd] to read `VCF` files into `R` and extract fields.

The example `VCF` files below were annotated with VEP. In order to extract VEP annotation, the file needs to be first parsed with `parse_vcf_output` and then with `parse_vep_csq`.

### `parse_vcf_output`

```{r include=TRUE,echo=TRUE}
annot_vcf_mutect <- system.file("extdata", "germline_mutect.vcf", package = "lineplots")
annot_vcf_varscan <- system.file("extdata", "germline_varscan.vcf", package = "lineplots")
annot_vcf_vardict <- system.file("extdata", "germline_vardict.vcf", package = "lineplots")

annot_vcf_mutect
```

```{r include=TRUE,echo=TRUE}
parsed_vcf_mutect <- parse_vcf_output(annot_vcf_mutect, caller = "mutect",sample_name = "Sample1")
parsed_vcf_varscan <- parse_vcf_output(annot_vcf_varscan, caller = "varscan",sample_name = "Sample1")
parsed_vcf_vardict <- parse_vcf_output(annot_vcf_vardict, caller = "vardict",sample_name = "Sample1")
```

```{r}
head(parsed_vcf_mutect)
```

```{r}
head(parsed_vcf_varscan)
```


```{r}
head(parsed_vcf_vardict)
```

### `parse_vep_csq`

Now we can parse the `CSQ` field added by the VEP.

```{r include=TRUE,echo=TRUE}
stand_out <- parse_vep_csq(vcf_path = annot_vcf_varscan, vcf_df = parsed_vcf_varscan)

knitr::kable(stand_out[1:10,c("Location","caller","Feature_type","IMPACT")],caption = "Parse VCF from VarScan and annotated with VEP.")
```


## Inputs of the `import` function

The two output formats can be nested lists or tidy data frames which can be used for different visualisation purposes. 

Below I am going to describe the usage of the import functions available in the package. 


## From `VCF` to suitable inputs for `import_any_for_lineplots()`

The function `import_any_for_lineplots()` is a general function 


## Vignette Info

Note the various macros within the `vignette` section of the metadata block above. These are required in order to instruct R how to build the vignette. Note that you should change the `title` field and the `\VignetteIndexEntry` to match the title of your vignette.

## Styles

The `html_vignette` template includes a basic CSS theme. To override this theme you can specify your own CSS in the document metadata as follows:

    output: 
      rmarkdown::html_vignette:
        css: mystyles.css

## Figures

The figure sizes have been customised so that you can easily put two images side-by-side. 

```{r, fig.show='hold'}
plot(1:10)
plot(10:1)
```

You can enable figure captions by `fig_caption: yes` in YAML:

    output:
      rmarkdown::html_vignette:
        fig_caption: yes

Then you can use the chunk option `fig.cap = "Your figure caption."` in **knitr**.

## More Examples

You can write math expressions, e.g. $Y = X\beta + \epsilon$, footnotes^[A footnote here.], and tables, e.g. using `knitr::kable()`.

```{r, echo=FALSE, results='asis'}
knitr::kable(head(mtcars, 10))
```

Also a quote using `>`:

> "He who gives up [code] safety for [code] speed deserves neither."
([via](https://twitter.com/hadleywickham/status/504368538874703872))
