---
title: "Introduction to varikondo"
author: "Anna Quaglieri"
date: "`r Sys.Date()`"
output:
  rmarkdown::html_vignette:
    fig_caption: yes
    toc_float: true
bibliography: ../biblio.bib
vignette: >
  %\VignetteIndexEntry{Vignette Title}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE,
                      message = FALSE,
                      warning = FALSE,
                      collapse = TRUE,
                      comment = "#>")
```


# Setup 

The `varikondo` R package comprises several helper and import functions that can be used to convert output from any of the callers mentioned above to a standardised format.

```{r eval=FALSE}
library(devtools)
devtools::install_github("annaquaglieri16/varikondo")
```


```{r eval=TRUE}
library(varikondo)
utils::packageVersion("varikondo")
```

```{r}
ls("package:varikondo")
```

# Import and parse a single `VCF` to tidy data frame

The function `parse_vcf_output()` allows to convert a `VCF` file generated by one of four callers [`GATK3 MuTect2`](https://software.broadinstitute.org/gatk/documentation/tooldocs/3.8-0/org_broadinstitute_gatk_tools_walkers_cancer_m2_MuTect2.php), [`VarDict`](https://github.com/AstraZeneca-NGS/VarDict), [`VarScan`](http://varscan.sourceforge.net/), and [`freebayes`](https://github.com/ekg/freebayes) to a data frame with standardised fields. This is because different callers annotate the `INFO` field in the `VCF` ouput in different ways, using different names for read depth, variant allele frequency etc... The output from `parse_vcf_output()` can then be used as input for `import_any()` to combine variants and clinical information for different samples within the same patients and to fill in missing variants at specific time points. If the variants in the `VCF` file are also annotated with the Variant Effect Predictor (VEP) [@McLaren2016-lv], the one can set `vep = TRUE` in `parse_vcf_output()` to parse the extra fields. 

It is to be noted that `VEP` might annotate the same variant multiple times, depending on whether a variants falls on several transcripts. VEP pastes together different annotations on the same line. `parse_vcf_output` will reshape the input variants and return it in a long format by stacking all the annotations one underneath each other. This has the potential to largely increase the size of the ouput file and it is suggested to subset the initial VCF file into chromosome if the inial one is too large.

The `parse_vcf_output` functions is built on top of the [`VariantAnnotation`](https://bioconductor.org/packages/release/bioc/html/VariantAnnotation.html) Bioconductor package [@Obenchain15072014] to quickly read large `VCF` files into `R` and extract fields. The article [Standardise VCFs output across callers](file:///Volumes/AML_RNA/quaglieri.a/R_packages/varikondo/docs/articles/standardise-variant-fields.html) describes what fields are extracted from a caller's `VCF` file to populate the joyful standardised fields given in ouput.



The example `VCF` files below were annotated with VEP. Also `chr20_varscan.vcf.gz`, `chr20_vardict.vcf.gz` are available as example. 

```{r include=TRUE,echo=TRUE}
annot_vcf_mutect <- system.file("extdata", "chr20_mutect.vcf.gz", package = "varikondo")
annot_vcf_mutect
```

`VCF` files from `VarDict`, `VarScan`, and `freebayes` can be parsed by specifying `caller = "vardict"`, `caller = "varscan"`, `caller = "freebayes"`.

```{r include=TRUE,echo=TRUE,eval=TRUE}
parsed_vcf_mutect <- varikondo::parse_vcf_output(vcf_path = annot_vcf_mutect, 
                                                 caller = "mutect",
                                                 sample_name = "Sample1",
                                                 vep = TRUE)
```

```{r parsed-vcf-mut}
knitr::kable(parsed_vcf_mutect[1:10,],caption = "Parsed MuTect2 output, from VCF to data frame.")
```


# Import functions

In general, the import functions can take in input variants called by a caller as well as clinical information for all the patients in the cohort under investigation. Every input function requires a `patientID` to be provided and a set of `studyGenes` to restrict the analysis on. `studygenes` are expected to be a character vector of NCBI gene `Symbols`. This means that variants need to be annotated with the gene they belong to before `varikondo` can be used. The clinical data at every time point is needed because if a variant is not detected at one time point for a patient, then the input variants will be extended to include all time points initially available for that patient but with default values for missing fields (e.g. VAF = 0). This is useful for visualisation purposes where one needs to track the development of mutation over time.


## Import, combine and standardise `superFreq` outputs

The `import_goi_superfreq()` function was specifically build to extract all the outputs produced by the caller which would be otherwise stored in nested lists within each patient's `Rdata` file or in `.csv` file if sometic variants. `superFreq` produces output from sometic SNVs, CNVs and clone tracking. A `superFreq` run for a cohort of patients (view [instructions](https://github.com/ChristofferFlensburg/superFreq)) will create and `R` folder with results for each patient, each stored within a folder. `import_goi_supefreq` will look into this folder to extract all the results. `import_goi_supefreq` also needs a path to the cohort metadata file in order to match the ouput from each sample with its patient name and time points.  

```{r eval=FALSE}
import_sf <- varikondo::import_goi_superfreq(superFreq_R_path = superFreq_R_path,
   superFreq_meta_path = superFreq_meta_path,
  patientID = patientID,
  studyGenes = "KIT"
  ref_genome = "hg38",
  min_vaf = 0.05,
  min_alt = 3)
```


### How different types of variants are provided in output

We define genetic lesions (SNVs, INDELs, CNAs) based on their `chrom`, `pos`, `SYMBOL`, `mutation_key`, `mutation_det`, `ref`, `alt` which are defined below. Different genetic lesions can have slightly different definitions of these features. `superFreq` reports SNVs, CNAs and clones which are simply clusters of the above mutations. Below are examples on how the different genetic lesions are reported. 

#### **Single Nucleotide Variants (SNVs)**
  
  - `chrom`: chromosome;
  - `pos`: position if point mutation in base pairs with reference to a genome build;
  - `SYMBOL`: `NCBI` gene symbol where the `SNV` occurs;
  - `ref`: reference allele; 
  - `alt`: alternative allele; 
  - `mutation_key`: unique key for the mutation defined as `SYMBOL-pos-ref-alt`. This is needed to summarise and look for the same mutation across different samples. 
  - `mutation_det`: details of the mutation defined as `SYMBOL Any annotation`. This can be used to annotate plots with features of a variant. Since `import_goi_superfreq` only imports output from `superFreq`, the annotation used to populate the `mutation_det` field for `SNVs` is the annotation provided by a variant annotation tool, like `VEP`. However, `mutation_det` can be updated at will to change plot labeling. This is also discussed below when the `import_any` function is introduced. 

```{r echo = FALSE}
snvs <- data.frame(PID = "P1",
                   chrom = "chr1", 
                   pos = 4573828,
                   SYMBOL = "KIT",
                   ref = "A",
                   alt = "G",
                   mutation_key = "KIT-4573828-A-G",
                   Consequence = "nonsynonimous",
                   mutation_det = "KIT nonsynonimous",
                   variant_type=c("SNV"))
knitr::kable(snvs)
```

We define genetic lesions based on: `chrom`, `pos`, `SYMBOL`, `mutation_key`, `mutation_det`, `ref`, `alt`. Different genetic lesions can have slightly different definitions of the features above. In particular:

#### **Copy Number Alterations (CNAs)** 

A CNA involves deletions of amplifications of genomic regions of different size. In this pipeline we use `superFreq` to call genomic alterations but the ouput could be adapted for other callers. `superFreq` reports a CNA specifying the `chrom` where the alteration occurs; the widths of the region involved (150Bbp in the example below); and the alteration type. For example, if `AB` represents the normal genotype, then `150Mbp A` a 150Mbp loss of one allele of a chromosome which involves genes KDM6A, SMC1A and BCORL1. Since, it is common practice to summarise mutations occurring on genes, the `mutation_key` is defined here as `SYMBOL-genotype`. This means that if plotting CNAs ocurring in one patient there will be multiple instances of the same CNA replecting the number of genes involved and when plotting mutations occurring on a specific gene there will only be one instance per gene. CNAs are not annotated by a variant annotation tools which is why the mutation details, `mutation_det`, is created as `SYMBOL width_of_CNA genotype`. 

```{r echo = FALSE}
cna <- data.frame(chrom = "chrX", 
                   pos = 276323,
                   SYMBOL = c("KDM6A","SMC1A","BCORL1"),
                   ref = "",
                   alt = "",
                   Consequence = "",
                   mutation_key = c("KDM6A-A","SMC1A-A","BCORL1-A"),
                   mutation_det = c("KDM6A 150Mbp A","SMC1A 150Mbp A","BCORL1 150Mbp A"),
                   variant_type=c("CNA","CNA","CNA"))
knitr::kable(cna)
```


#### **Clones** 

In `superFreq` clones are collections of SNVs and CNAs that behave the same over time. Therefore, they won't have a specific notation apart from identifiying how much a clone changes over time (using an estimate of `clonality`, similar to `VAF`) and how many events are involved in each clone. `varikondo` will be updated so that it will be possible to extract the genes involved in every clones as compyted by `superFreq`. 

Within every patient there will be a finite number of clones identified, 3 in the example below as sumamrised by `mutation_key`. `mutation_det` reports how many `anchors` (events) are in each clone identified. 

```{r echo = FALSE}
clone <- data.frame(PID = c("P1","P1","P1"),
  chrom = "", 
                   pos = "",
                   SYMBOL = "",
                   ref = "",
                   alt = "",
                   mutation_key = c("1","2","3"),
                   mutation_det = c("clone (4 anchors)","clone (10 anchors)","clone (7 anchors)"),
                   variant_type=c("clones","clones","clones"))
knitr::kable(clone)
```


#### **INDELs** 
  
  - `chrom`, `pos`, `SYMBOL`, `ref`, `alt`, `mutation_key` are the same as for `SNVs`.
  - `mutation_det`: details of the mutation defined as `SYMBOL-Consequence`. If the `Consequence` from a variant annotation tool is not available then this can be for example the exon where the variant occurs. In the example below `Consequence` was defined as `ITD` which stands for `Internal Tandem Duplication` which specifies the type of INDEL occurring in the FLT3 gene. 

```{r echo = FALSE}
indel <- data.frame(chrom = "chr13", 
                   pos = 764739898,
                   SYMBOL = "FLT3",
                   ref = "",
                   alt = "GATGATGAT",
                   mutation_key = "chr13-764739898- -GATGATGAT",
                   Consequence = "ITD exon13",
                   mutation_det = "FLT3-ITD exon13",
                   variant_type=c("INDEL"))
knitr::kable(indel)
```


## `import_any()`

# Aknowledgements

