<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>A package to declutter, organise and standardise genomic variant calls • varikondo</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.3.7/sandstone/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="A package to declutter, organise and standardise genomic variant calls">
<meta property="og:description" content="The package was developed to organise, standardise and combine variant calling output from several callers for a cohort of patients and for genes of interest.">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">varikondo</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.6.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="articles/vignette.html">Get started</a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="articles/standardise-variant-fields.html">Standardise VCF fields across callers</a>
    </li>
    <li>
      <a href="articles/how-variants-are-stored.html">How different types of variants are stored</a>
    </li>
    <li>
      <a href="articles/rna-seq-pipe.html">Example - Variant calling in RNA-Seq</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/annaquaglieri16/varikondo">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    

    
    
<div id="varikondo-find-the-spark-joy-in-your-variants" class="section level1">
<div class="page-header"><h1 class="hasAnchor">
<a href="#varikondo-find-the-spark-joy-in-your-variants" class="anchor"></a>varikondo: Find the spark joy in your variants!</h1></div>
<p>Bioinformatics can be very overwhelming at times, with all its different file formats, fields specifications and many ways of storing the same things! That’s how the <code>varikondo</code> package came to life. It was developed by the need to <strong>re-organise</strong>, <strong>standardise</strong> and <strong>combine</strong> variant calling outputs from several callers.</p>
</div>
<div id="install" class="section level1">
<h1 class="hasAnchor">
<a href="#install" class="anchor"></a>Install</h1>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(devtools)</a>
<a class="sourceLine" id="cb1-2" data-line-number="2">devtools<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/devtools/topics/reexports">install_github</a></span>(<span class="st">"annaquaglieri16/varikondo"</span>)</a>
<a class="sourceLine" id="cb1-3" data-line-number="3"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(varikondo)</a></code></pre></div>
</div>
<div id="main-functionalities" class="section level1">
<h1 class="hasAnchor">
<a href="#main-functionalities" class="anchor"></a>Main functionalities</h1>
<div id="import-variants-from-several-callers" class="section level2">
<h2 class="hasAnchor">
<a href="#import-variants-from-several-callers" class="anchor"></a>Import variants from several callers</h2>
<p>Variants from different callers can be imported in R and parsed into a data frame with standardised fields. See <a href="https://annaquaglieri16.github.io/varikondo/articles/standardise-variant-fields.html">how fields are standardaised</a> to know more about which fields from each callers are parsed. The functions which achieve this are:</p>
<ul>
<li>
<code><a href="reference/import_goi_superfreq.html">import_goi_superfreq()</a></code>: to import variants found on genes of interest called by <a href="https://github.com/ChristofferFlensburg/superFreq">superFreq</a>, a caller that allows clonal tracking of cancer sequencing data. Figure 1 sketches how the function works.</li>
</ul>
<div class="figure">
<img src="reference/figures/import-superfreq.png" alt="Figure 1. Overview of the functionality to import and organise superFreq outputs." width="621"><p class="caption">
Figure 1. Overview of the functionality to import and organise superFreq outputs.
</p>
</div>
<ul>
<li>
<code><a href="reference/parse_vcf_output.html">parse_vcf_output()</a></code>: which takes a <a href="http://www.internationalgenome.org/wiki/Analysis/Variant%20Call%20Format/vcf-variant-call-format-version-40/">Variant Call Format (VCF)</a> file as input and outputs a parsed data frame with standardised names for the fields in the <code>VCF</code> file. Currently, <code><a href="reference/parse_vcf_output.html">parse_vcf_output()</a></code> can parse ouputs from the following callers run in tumour-only mode: <code>GATK3 MuTect2</code>, <code>VarScan2</code>, <code>VarDict</code> and <code>Freebayes</code>. It also enables parsing of the annotation field added by the <a href="https://asia.ensembl.org/info/docs/tools/vep/index.html">Variant Effect Predictor</a>.</li>
</ul>
<p>See more examples in <a href="https://annaquaglieri16.github.io/varikondo/articles/vignette.html"><code>Get Started</code></a>.</p>
</div>
<div id="combine-variant-calls-from-different-samples-within-the-same-patient" class="section level2">
<h2 class="hasAnchor">
<a href="#combine-variant-calls-from-different-samples-within-the-same-patient" class="anchor"></a>Combine variant calls from different samples within the same patient</h2>
<ul>
<li>
<code><a href="reference/import_any.html">import_any()</a></code> combines and filters the variant calls (already imported into <code>R</code>) from several samples collected for the same patient. This functions attempts to mimic in a very simplistic way what is performed within <a href="https://github.com/ChristofferFlensburg/superFreq">superFreq</a>. Instead of filtering each variant callset separately for each sample, variants are filtered jointly by patient. This means that if a variant reaches minimal quality threshold at any time point, then it will be kept for all other time points. If a variant is not detected in some samples it will be filled with default values, e.g. <code>zero</code> variant allele frequency. This allows to have complete information for a variant across time for one patient and to plot and evaluate the shift in VAF or depth over time.</li>
</ul>
<p>Below is a graphic overview of how the output of <code><a href="reference/parse_vcf_output.html">parse_vcf_output()</a></code> and <code><a href="reference/import_any.html">import_any()</a></code> interact. The example is based on files for one patient but the same can be extended to any patient.</p>
<div class="figure">
<img src="reference/figures/import-any.png" alt="Figure 2. Overview of the functionality to import and organise your VCF files." width="624"><p class="caption">
Figure 2. Overview of the functionality to import and organise your VCF files.
</p>
</div>
<p>See more examples in <a href="https://annaquaglieri16.github.io/varikondo/articles/vignette.html"><code>Get Started</code></a>.</p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <div class="links">
<h2>Links</h2>
<ul class="list-unstyled">
<li>Browse source code at <br><a href="https://github.com/annaquaglieri16/varikondo">https://​github.com/​annaquaglieri16/​varikondo</a>
</li>
<li>Report a bug at <br><a href="https://github.com/annaquaglieri16/varikondo/issues">https://​github.com/​annaquaglieri16/​varikondo/​issues</a>
</li>
</ul>
</div>
<div class="license">
<h2>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a></small></li>
</ul>
</div>
<div class="developers">
<h2>Developers</h2>
<ul class="list-unstyled">
<li>
<a href="https://github.com/annaquaglieri16">Anna Quaglieri</a> <br><small class="roles"> Author, maintainer </small>  </li>
<li>
<a href="https://github.com/ChristofferFlensburg">Christoffer Flensburg</a> <br><small class="roles"> Author, contributor </small>  </li>
</ul>
</div>

      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by <a href="https://github.com/annaquaglieri16">Anna Quaglieri</a>, <a href="https://github.com/ChristofferFlensburg">Christoffer Flensburg</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.9000.</p>
</div>
      </footer>
</div>

  

  </body>
</html>
