<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>How different types of variants are stored â€¢ varikondo</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.3.7/sandstone/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="How different types of variants are stored">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">varikondo</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.6.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/vignette.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/standardise-variant-fields.html">Standardise VCF fields across callers</a>
    </li>
    <li>
      <a href="../articles/how-variants-are-stored.html">How different types of variants are stored</a>
    </li>
    <li>
      <a href="../articles/rna-seq-pipe.html">Example - Variant calling in RNA-Seq</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/annaquaglieri16/varikondo">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>How different types of variants are stored</h1>
                        <h4 class="author">Anna Quaglieri</h4>
            
            <h4 class="date">2019-06-18</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/annaquaglieri16/varikondo/blob/master/vignettes/how-variants-are-stored.Rmd"><code>vignettes/how-variants-are-stored.Rmd</code></a></small>
      <div class="hidden name"><code>how-variants-are-stored.Rmd</code></div>

    </div>

    
    
<div id="how-we-store-different-types-of-variants-in-output" class="section level1">
<h1 class="hasAnchor">
<a href="#how-we-store-different-types-of-variants-in-output" class="anchor"></a>How we store different types of variants in output</h1>
<p>We define genetic lesions (SNVs, INDELs, CNAs) based on their <code>chrom</code>, <code>pos</code>, <code>SYMBOL</code>, <code>mutation_key</code>, <code>mutation_det</code>, <code>ref</code>, <code>alt</code> defined below. Different genetic lesions can have slightly different definitions of these features. <code>superFreq</code> reports SNVs, short INDELs, CNAs and clones. Below are examples on how the fields above are defined for each genetic lesion.</p>
<div id="single-nucleotide-variants-snvs-and-short-indels-reported-by-superfreq" class="section level2">
<h2 class="hasAnchor">
<a href="#single-nucleotide-variants-snvs-and-short-indels-reported-by-superfreq" class="anchor"></a>Single Nucleotide Variants (SNVs) and short INDELs reported by superFreq</h2>
<ul>
<li>
<code>chrom</code>: chromosome;</li>
<li>
<code>pos</code>: position of point mutation in base pairs, with reference to a genome build;</li>
<li>
<code>SYMBOL</code>: <code>NCBI</code> gene symbol where the <code>SNV</code> occurs;</li>
<li>
<code>ref</code>: reference allele;</li>
<li>
<code>alt</code>: alternative allele;</li>
<li>
<code>mutation_key</code>: unique key for the mutation usually defined as <code>SYMBOL-pos-ref-alt</code>. This is needed to summarise and look for the same mutation across different samples.</li>
<li>
<code>mutation_det</code>: details of the mutation usually defined as <code>SYMBOL Any annotation</code>. This can be used to annotate plots with features of a variant. Since <code><a href="../reference/import_goi_superfreq.html">import_goi_superfreq()</a></code> only imports output from <code>superFreq</code>, the annotation used to populate the <code>mutation_det</code> field for <code>SNVs</code> is the annotation provided by a variant annotation tool, like <a href="https://asia.ensembl.org/info/docs/tools/vep/index.html"><code>VEP</code></a> or [<code>VariantAnnotation</code>](<a href="https://bioconductor.org/packages/release/bioc/html/VariantAnnotation.html" class="uri">https://bioconductor.org/packages/release/bioc/html/VariantAnnotation.html</a>. However, <code>mutation_det</code> can be updated at will to change plot labeling.</li>
</ul>
<table class="table">
<thead><tr class="header">
<th align="left">PID</th>
<th align="left">chrom</th>
<th align="right">pos</th>
<th align="left">SYMBOL</th>
<th align="left">ref</th>
<th align="left">alt</th>
<th align="left">mutation_key</th>
<th align="left">Consequence</th>
<th align="left">mutation_det</th>
<th align="left">variant_type</th>
</tr></thead>
<tbody><tr class="odd">
<td align="left">P1</td>
<td align="left">chr1</td>
<td align="right">4573828</td>
<td align="left">KIT</td>
<td align="left">A</td>
<td align="left">G</td>
<td align="left">KIT-4573828-A-G</td>
<td align="left">nonsynonimous</td>
<td align="left">KIT nonsynonimous</td>
<td align="left">SNV</td>
</tr></tbody>
</table>
</div>
<div id="copy-number-alterations-cnas" class="section level2">
<h2 class="hasAnchor">
<a href="#copy-number-alterations-cnas" class="anchor"></a>Copy Number Alterations (CNAs)</h2>
<p>A CNA involves deletions or amplifications of genomic regions of different sizes. In this pipeline we use <code>superFreq</code> to call genomic alterations but the ouput could be adapted for other callers. <code>superFreq</code> reports a CNA specifying the <code>chrom</code> where the alteration occurs; the widths of the region involved (150Bbp in the example below); and the alteration type. For example, if <code>AB</code> represents the normal genotype, then <code>150Mbp A</code> is a 150Mbp loss of one allele. Since, it is common practice to summarise mutations occurring on genes, the <code>mutation_key</code> is defined here as <code>SYMBOL-genotype</code>. This means that when plotting CNAs for one patient, there could be multiple instances of the same CNA reflecting the number of <code>studyGenes</code> involved in that CNA. CNAs are not annotated by a variant annotation tool which is why the mutation details, <code>mutation_det</code>, is simply <code>SYMBOL width_of_CNA genotype</code>.</p>
<table class="table">
<thead><tr class="header">
<th align="left">chrom</th>
<th align="right">pos</th>
<th align="left">SYMBOL</th>
<th align="left">ref</th>
<th align="left">alt</th>
<th align="left">Consequence</th>
<th align="left">mutation_key</th>
<th align="left">mutation_det</th>
<th align="left">variant_type</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">chrX</td>
<td align="right">276323</td>
<td align="left">KDM6A</td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left">KDM6A-A</td>
<td align="left">KDM6A 150Mbp A</td>
<td align="left">CNA</td>
</tr>
<tr class="even">
<td align="left">chrX</td>
<td align="right">276323</td>
<td align="left">SMC1A</td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left">SMC1A-A</td>
<td align="left">SMC1A 150Mbp A</td>
<td align="left">CNA</td>
</tr>
<tr class="odd">
<td align="left">chrX</td>
<td align="right">276323</td>
<td align="left">BCORL1</td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left">BCORL1-A</td>
<td align="left">BCORL1 150Mbp A</td>
<td align="left">CNA</td>
</tr>
</tbody>
</table>
</div>
<div id="clones" class="section level2">
<h2 class="hasAnchor">
<a href="#clones" class="anchor"></a>Clones</h2>
<p>In <code>superFreq</code> clones are collections of SNVs and CNAs that change together over time. Therefore, they wonâ€™t have a specific notation apart from identifiying how much a clone changes over time (using an estimate of <code>clonality</code>, similar to <code>VAF</code>) and how many events are involved in each clone. <code>varikondo</code> will be updated so that it will be possible to extract the genes involved in every clones as compyted by <code>superFreq</code>.</p>
<p>Within every patient there will be a finite number of clones identified (3 in the example below) as sumamrised by <code>mutation_key</code>. <code>mutation_det</code> reports how many <code>anchors</code> (events) are in each clone identified.</p>
<table class="table">
<thead><tr class="header">
<th align="left">PID</th>
<th align="left">chrom</th>
<th align="left">pos</th>
<th align="left">SYMBOL</th>
<th align="left">ref</th>
<th align="left">alt</th>
<th align="left">mutation_key</th>
<th align="left">mutation_det</th>
<th align="left">variant_type</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">P1</td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left">1</td>
<td align="left">clone (4 anchors)</td>
<td align="left">clones</td>
</tr>
<tr class="even">
<td align="left">P1</td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left">2</td>
<td align="left">clone (10 anchors)</td>
<td align="left">clones</td>
</tr>
<tr class="odd">
<td align="left">P1</td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left">3</td>
<td align="left">clone (7 anchors)</td>
<td align="left">clones</td>
</tr>
</tbody>
</table>
</div>
<div id="indels-reported-by-vardict" class="section level2">
<h2 class="hasAnchor">
<a href="#indels-reported-by-vardict" class="anchor"></a>INDELs reported by VarDict</h2>
<ul>
<li>
<code>chrom</code>, <code>pos</code>, <code>SYMBOL</code>, <code>ref</code>, <code>alt</code>, <code>mutation_key</code> are the same as for <code>SNVs</code>.</li>
<li>
<code>mutation_det</code>: details of the mutation defined as <code>SYMBOL-Consequence</code>. If the <code>Consequence</code> from a variant annotation tool is not available then this can be for example the exon where the variant occurs. In the example below <code>Consequence</code> was defined as <code>ITD</code> which stands for <code>Internal Tandem Duplication</code> which specifies the type of INDEL occurring in the FLT3 gene.</li>
</ul>
<table class="table">
<thead><tr class="header">
<th align="left">chrom</th>
<th align="right">pos</th>
<th align="left">SYMBOL</th>
<th align="left">ref</th>
<th align="left">alt</th>
<th align="left">mutation_key</th>
<th align="left">Consequence</th>
<th align="left">mutation_det</th>
<th align="left">variant_type</th>
</tr></thead>
<tbody><tr class="odd">
<td align="left">chr13</td>
<td align="right">764739898</td>
<td align="left">FLT3</td>
<td align="left"></td>
<td align="left">GATGATGAT</td>
<td align="left">chr13-764739898- -GATGATGAT</td>
<td align="left">ITD exon13</td>
<td align="left">FLT3-ITD exon13</td>
<td align="left">INDEL</td>
</tr></tbody>
</table>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li>
<a href="#how-we-store-different-types-of-variants-in-output">How we store different types of variants in output</a><ul class="nav nav-pills nav-stacked">
<li><a href="#single-nucleotide-variants-snvs-and-short-indels-reported-by-superfreq">Single Nucleotide Variants (SNVs) and short INDELs reported by superFreq</a></li>
      <li><a href="#copy-number-alterations-cnas">Copy Number Alterations (CNAs)</a></li>
      <li><a href="#clones">Clones</a></li>
      <li><a href="#indels-reported-by-vardict">INDELs reported by VarDict</a></li>
      </ul>
</li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by <a href="https://github.com/annaquaglieri16">Anna Quaglieri</a>, <a href="https://github.com/ChristofferFlensburg">Christoffer Flensburg</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.9000.</p>
</div>
      </footer>
</div>

  

  </body>
</html>
