<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Standardise VCF fields across callers â€¢ varikondo</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.3.7/sandstone/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Standardise VCF fields across callers">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">varikondo</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.3.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/vignette.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../articles/standardise-variant-fields.html">Articles</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/annaquaglieri16/varikondo">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Standardise VCF fields across callers</h1>
                        <h4 class="author">Anna Quaglieri</h4>
            
      
      <small class="dont-index">Source: <a href="https://github.com/annaquaglieri16/varikondo/blob/master/vignettes/standardise-variant-fields.Rmd"><code>vignettes/standardise-variant-fields.Rmd</code></a></small>
      <div class="hidden name"><code>standardise-variant-fields.Rmd</code></div>

    </div>

    
    
<p>Every caller has its own ways of naming <code>VCF</code> fields describing variant information. Following, is the list of fields which are given as output after parsing a VCF file with <code><a href="../reference/parse_vcf_output.html">parse_vcf_output()</a></code>. Here is a description of what information was extracted for every caller to populate each standardised fields:</p>
<ul>
<li>
<em>Location</em>: <code>CHR_POS</code> where <code>CHR</code> and <code>POS</code> are standard <code>VCF</code> fields for every caller.</li>
<li>
<em>caller</em>: the caller passed to the <code>caller</code> argument.</li>
<li>
<em>chrom</em>: field <code>CHROM</code> in the <code>VCF</code> file.</li>
<li>
<em>pos</em>: field <code>POS</code> in the <code>VCF</code> file.</li>
<li>
<em>ref</em>: field <code>REF</code> in the <code>VCF</code> file.</li>
<li>
<em>alt</em>: field <code>ALT</code> in the <code>VCF</code> file.</li>
<li>
<em>qual</em>: This field is not reported consistently from the VCF output of the three callers but I made sure that its meaning is consistent and that it represents the average base quality at that position. Below is a description of how it is extracted from each caller.
<ul>
<li>In <code>MuTect2</code> the <code>QSS</code> field in the <code>FORMAT</code> fields reports the sum of the base qualities for the reference and alternative alleles separated by a comma. We used the reference and alternative depths at each position to compute the overall average base quality at that position. This quantity will populate the final <code>qual</code> field.</li>
<li>In <code>VarScan</code> this field is the average of the <code>RBQ</code> and <code>ABQ</code> fields from the <code>FORMAT</code> fields. They are defined respectively as the average quality of reference and alternative supporting bases in the header of the <code>VCF</code> file.</li>
<li>In <code>VarDict</code> there are two <code>QUAL</code> fields, one is the standard 6th field of a <code>VCF</code> file and the other one is reported in the <code>INFO</code> fields. We will use the latter to populate the <code>qual</code> field in our analysis since it is the one defined as the average base quality at a variant position in the header of the <code>VCF</code>. <code>VarDict</code> uses a threshold of <code>QUAL &gt;= 25</code> to report a variant.</li>
</ul>
</li>
<li><p><em>filter</em>: standard field <code>FILTER</code> in the <code>VCF</code> file. Each caller populates this field in different ways depending on the characteristics of the algorithm. In general, the entries for this field can be either <code>PASS</code> if that mutation passes all the filters defined by a caller or a description of the reason for filtering. The possible descriptions and their meaning can be found in the header of the VCF file generated by the caller.</p></li>
<li><p><em>genotype</em>: standard <code>GT</code> field in the <code>FORMAT</code> fields of the <code>VCF</code> file.</p></li>
<li><p><em>tot_depth</em>: total read depth at each position as estimated by the caller. This information can be reported differently by each caller. <code>VarDict</code> and <code>VarScan</code> record it in the <code>DP</code> field while <code>MuTect2</code> records the reference and alternative depth in the <code>AD</code> columns and their sum was used to define the <code>total_depth</code>.</p></li>
<li><p><em>VAF</em>: variant allele frequency for the variants recorded at that position. <code>VarScan</code> records it in the <code>FREQ</code> field while <code>MuTect</code> and <code>VarDict</code> in the <code>AF</code> field.</p></li>
<li><p><em>ADJVAF_ADJ_indels</em>: field <code>ADJAF</code> only reported by <code>VarDict</code> and it represents the adjusted variant allele frequency for indels due to local realignment.</p></li>
<li>
<em>ref_depth</em>, <em>alt_depth</em>, <em>ref_forw</em>, <em>ref_rev</em>, <em>alt_forw</em> and <em>alt_rev</em>: these fields represent the breakdown of supporting reference/alternative and forward/reverse reads at each location. Below I described what fields I used from every caller to extract these values. The fields are listed in order:
<ul>
<li>
<code>MuTect2</code>: <em>ref_depth</em> and <em>alt_depth</em> are the comma separated values reported in the field <code>AD</code>; <em>ref_forw</em>, <em>ref_rev</em>, <em>alt_forw</em> and <em>alt_rev</em> are respectively the <code>MuTect2</code> fields <code>REF_F1R2</code>, <code>REF_F2R1</code>, <code>ALT_F1R2</code> and <code>ALT_F2R1</code>.</li>
<li>
<code>VarScan2</code>: in order the features listed above are extracted from the the fields <code>RD</code>, <code>AD</code>, <code>RDF</code>, <code>RDR</code>, <code>ADF</code> and <code>ADR</code>.</li>
<li>
<code>VarDict</code>: <em>ref_depth</em> and <em>alt_depth</em> are the fields <code>REF</code> and <code>ALT</code> in the VCF file; the field <code>REFBIAS</code> contains comma separated values representing <em>ref_forw</em> and <em>ref_rev</em> and <code>VARBIAS</code> contains comma separated values representing <em>alt_forw</em> and <em>alt_rev</em>.</li>
</ul>
</li>
</ul>
<p>If the <code>VCF</code> file was also annotated using the Variant Effect Predictor (VEP) <span class="citation">[@McLaren2016-lv]</span><br>
All the other fields are exactly as generated by <code>VEP</code>: <code>Allele</code>, <code>Consequence</code>, <code>IMPACT</code>, <code>SYMBOL</code>, <code>Gene</code>, <code>Feature_type</code>, <code>Feature</code>, <code>BIOTYPE</code>, <code>EXON</code>, <code>INTRON</code>, <code>HGVSc</code>, <code>HGVSp</code>, <code>cDNA_position</code>, <code>CDS_position</code>, <code>Protein_position</code> <code>Amino_acids</code>, <code>Codons</code>, <code>Existing_variation</code> <code>DISTANCE</code>, <code>STRAND</code>, <code>FLAGS</code>, <code>VARIANT_CLASS</code>, <code>SYMBOL_SOURCE</code>, <code>HGNC_ID</code>, <code>CANONICAL</code>, <code>TSL</code>, <code>APPRIS</code>, <code>CCDS</code>, <code>ENSP</code>, <code>SWISSPROT</code>, <code>TREMBL</code>, <code>UNIPARC</code>, <code>GENE_PHENO</code>, <code>SIFT</code>, <code>PolyPhen</code>, <code>DOMAINS</code>, <code>AF</code>, <code>AFR_AF</code>, <code>AMR_AF</code>, <code>EAS_AF</code>, <code>EUR_AF</code>, <code>SAS_AF</code>, <code>AA_AF</code>, <code>EA_AF</code>, <code>ExAC_AF</code>, <code>ExAC_Adj_AF</code>, <code>ExAC_AFR_AF</code>, <code>ExAC_AMR_AF</code>, <code>ExAC_EAS_AF</code>, <code>ExAC_FIN_AF</code>, <code>ExAC_NFE_AF</code>, <code>ExAC_OTH_AF</code>, <code>ExAC_SAS_AF</code>, <code>MAX_AF</code>, <code>MAX_AF_POPS</code>, <code>CLIN_SIG</code>, <code>SOMATIC</code>, <code>PHENO</code>, <code>PUBMED</code>,<code>MOTIF_NAME</code>, <code>MOTIF_POS</code>, <code>HIGH_INF_POS</code>, <code>MOTIF_SCORE_CHANGE</code> <code>SampleName</code>, <code>IMPACT_rank</code>. Visit the <code>VEP</code> page to find more information <a href="https://asia.ensembl.org/info/docs/tools/vep/index.html" class="uri">https://asia.ensembl.org/info/docs/tools/vep/index.html</a>.</p>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Anna Quaglieri, Christoffer Flensburg.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.9000.</p>
</div>
      </footer>
</div>

  

  </body>
</html>
