<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Introduction to varikondo • varikondo</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.3.7/sandstone/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Introduction to varikondo">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">varikondo</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.6.8</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/vignette.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/standardise-variant-fields.html">Standardise VCF fields across callers</a>
    </li>
    <li>
      <a href="../articles/how-variants-are-stored.html">How different types of variants are stored</a>
    </li>
    <li>
      <a href="../articles/rna-seq-pipe.html">Example - Variant calling in RNA-Seq</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/annaquaglieri16/varikondo">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Introduction to varikondo</h1>
                        <h4 class="author">Anna Quaglieri</h4>
            
            <h4 class="date">2019-07-15</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/annaquaglieri16/varikondo/blob/master/vignettes/vignette.Rmd"><code>vignettes/vignette.Rmd</code></a></small>
      <div class="hidden name"><code>vignette.Rmd</code></div>

    </div>

    
    
<div id="setup" class="section level1">
<h1 class="hasAnchor">
<a href="#setup" class="anchor"></a>Setup</h1>
<p><code>varikondo</code> comprises a suite of functions to convert output from purpuses superFreq <span class="citation">(Flensburg et al. 2018)</span>, VarScan <span class="citation">(Koboldt et al. 2012)</span>, VarDict <span class="citation">(Lai et al. 2016)</span>, GATK3 Mutect2 <span class="citation">(Cibulskis et al. 2013)</span>, and FreeBayes <span class="citation">(Garrison and Marth 2012)</span> into standardised data frames into R.</p>
<p>First, let’s load the package.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(devtools)</a>
<a class="sourceLine" id="cb1-2" data-line-number="2">devtools<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/devtools/topics/reexports">install_github</a></span>(<span class="st">"annaquaglieri16/varikondo"</span>)</a></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(varikondo)</a>
<a class="sourceLine" id="cb2-2" data-line-number="2">utils<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/packageDescription">packageVersion</a></span>(<span class="st">"varikondo"</span>)</a>
<a class="sourceLine" id="cb2-3" data-line-number="3"><span class="co">#&gt; [1] '0.6.8'</span></a></code></pre></div>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/ls">ls</a></span>(<span class="st">"package:varikondo"</span>)</a>
<a class="sourceLine" id="cb3-2" data-line-number="2"><span class="co">#&gt; [1] "combine_and_filter"   "example_metadata"     "example_variants"    </span></a>
<a class="sourceLine" id="cb3-3" data-line-number="3"><span class="co">#&gt; [4] "import_clinical"      "import_goi_superfreq" "parse_vcf_output"</span></a></code></pre></div>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(dplyr)</a>
<a class="sourceLine" id="cb4-2" data-line-number="2"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(ggplot2)</a>
<a class="sourceLine" id="cb4-3" data-line-number="3"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(cowplot)</a></code></pre></div>
</div>
<div id="import-and-parse-a-single-vcf" class="section level1">
<h1 class="hasAnchor">
<a href="#import-and-parse-a-single-vcf" class="anchor"></a>Import and parse a single <code>VCF</code>
</h1>
<p>The function <code><a href="../reference/parse_vcf_output.html">parse_vcf_output()</a></code> allows to convert a <code>VCF</code> file generated by one of four callers <a href="https://software.broadinstitute.org/gatk/documentation/tooldocs/3.8-0/org_broadinstitute_gatk_tools_walkers_cancer_m2_MuTect2.php"><code>GATK3 MuTect2</code></a>, <a href="https://github.com/AstraZeneca-NGS/VarDict"><code>VarDict</code></a>, <a href="http://varscan.sourceforge.net/"><code>VarScan</code></a>, and <a href="https://github.com/ekg/freebayes"><code>FreeBayes</code></a> into a data frame with standardised fields. This is because different callers annotate the <code>INFO</code> field in the <code>VCF</code> ouput in different ways, using different names for read depth, variant allele frequency etc… The output from <code><a href="../reference/parse_vcf_output.html">parse_vcf_output()</a></code> can then be used as input for <code><a href="../reference/combine_and_filter.html">combine_and_filter()</a></code> to combine variants and clinical information for different samples within the same patients and to fill in missing variants at specific time points. If the variants in the <code>VCF</code> file are also annotated with the Variant Effect Predictor (VEP) <span class="citation">(McLaren et al. 2016)</span>, <code>vep = TRUE</code> will parse the extra fields.</p>
<div id="import-only-specific-ranges-of-the-vcf-file" class="section level2">
<h2 class="hasAnchor">
<a href="#import-only-specific-ranges-of-the-vcf-file" class="anchor"></a>Import only specific ranges of the VCF file</h2>
<p>VEP might annotate the same variant multiple times, depending on whether a variants falls on several transcripts. VEP pastes together different annotations on the same line. <code><a href="../reference/parse_vcf_output.html">parse_vcf_output()</a></code> will reshape the input VCF and return it in a long format by stacking all the annotations one underneath each other. This has the potential to largely increase the size of the ouput file. To overcome this problem it is possible to read in R only restricted genomic regions at one time by specifying a <code>GRanges()</code> object in <code><a href="../reference/parse_vcf_output.html">parse_vcf_output(param = VariantAnnotation::ScanVcfParam())</a></code>. The <code>param</code> argument is directly passed to <code><a href="https://www.rdocumentation.org/packages/VariantAnnotation/topics/readVcf-methods">VariantAnnotation::readVcf()</a></code> within <code><a href="../reference/parse_vcf_output.html">parse_vcf_output()</a></code>.</p>
<p>The example <code>VCF</code> files below were annotated with VEP. Also <code>chr20_varscan.vcf.gz</code>, <code>chr20_vardict.vcf.gz</code> are available as example.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1">annot_vcf_mutect &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/system.file">system.file</a></span>(<span class="st">"extdata"</span>, <span class="st">"chr20_mutect.vcf.gz"</span>, <span class="dt">package =</span> <span class="st">"varikondo"</span>)</a>
<a class="sourceLine" id="cb5-2" data-line-number="2">annot_vcf_mutect</a>
<a class="sourceLine" id="cb5-3" data-line-number="3"><span class="co">#&gt; [1] "/Users/quaglieri.a/Library/R/3.5/library/varikondo/extdata/chr20_mutect.vcf.gz"</span></a></code></pre></div>
<p><code>VCF</code> files from <code>VarDict</code>, <code>VarScan</code>, and <code>freebayes</code> can be parsed by specifying <code>caller = "vardict"</code>, <code>caller = "varscan"</code>, <code>caller = "freebayes"</code>.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1">parsed_vcf_mutect &lt;-<span class="st"> </span>varikondo<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/varikondo/topics/parse_vcf_output">parse_vcf_output</a></span>(<span class="dt">vcf_path =</span> annot_vcf_mutect, </a>
<a class="sourceLine" id="cb6-2" data-line-number="2">                                                 <span class="dt">caller =</span> <span class="st">"mutect"</span>,</a>
<a class="sourceLine" id="cb6-3" data-line-number="3">                                                 <span class="dt">sample_name =</span> <span class="st">"Sample1"</span>,</a>
<a class="sourceLine" id="cb6-4" data-line-number="4">                                                 <span class="dt">vep =</span> <span class="ot">TRUE</span>)</a></code></pre></div>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1">knitr<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/knitr/topics/kable">kable</a></span>(parsed_vcf_mutect[<span class="dv">1</span><span class="op">:</span><span class="dv">10</span>,],<span class="dt">caption =</span> <span class="st">"Parsed MuTect2 output, from VCF to data frame."</span>)</a></code></pre></div>
<table class="table">
<caption>Parsed MuTect2 output, from VCF to data frame.</caption>
<thead><tr class="header">
<th align="left">Location</th>
<th align="left">caller</th>
<th align="left">chrom</th>
<th align="right">pos</th>
<th align="right">end</th>
<th align="left">ref</th>
<th align="left">alt</th>
<th align="right">qual</th>
<th align="left">filter</th>
<th align="left">genotype</th>
<th align="right">tot_depth</th>
<th align="right">VAF</th>
<th align="right">ref_depth</th>
<th align="right">alt_depth</th>
<th align="right">ref_forw</th>
<th align="right">ref_rev</th>
<th align="right">alt_forw</th>
<th align="right">alt_rev</th>
<th align="right">start</th>
<th align="right">width</th>
<th align="right">ref_base_quality</th>
<th align="right">alt_base_quality</th>
<th align="left">SampleName</th>
<th align="left">Allele</th>
<th align="left">Consequence</th>
<th align="left">IMPACT</th>
<th align="left">SYMBOL</th>
<th align="left">Gene</th>
<th align="left">Feature_type</th>
<th align="left">Feature</th>
<th align="left">BIOTYPE</th>
<th align="left">EXON</th>
<th align="left">INTRON</th>
<th align="left">HGVSc</th>
<th align="left">HGVSp</th>
<th align="left">cDNA_position</th>
<th align="left">CDS_position</th>
<th align="left">Protein_position</th>
<th align="left">Amino_acids</th>
<th align="left">Codons</th>
<th align="left">Existing_variation</th>
<th align="left">DISTANCE</th>
<th align="left">STRAND</th>
<th align="left">FLAGS</th>
<th align="left">VARIANT_CLASS</th>
<th align="left">SYMBOL_SOURCE</th>
<th align="left">HGNC_ID</th>
<th align="left">CANONICAL</th>
<th align="left">TSL</th>
<th align="left">APPRIS</th>
<th align="left">CCDS</th>
<th align="left">ENSP</th>
<th align="left">SWISSPROT</th>
<th align="left">TREMBL</th>
<th align="left">UNIPARC</th>
<th align="left">GENE_PHENO</th>
<th align="left">SIFT</th>
<th align="left">PolyPhen</th>
<th align="left">DOMAINS</th>
<th align="left">AF</th>
<th align="left">AFR_AF</th>
<th align="left">AMR_AF</th>
<th align="left">EAS_AF</th>
<th align="left">EUR_AF</th>
<th align="left">SAS_AF</th>
<th align="left">AA_AF</th>
<th align="left">EA_AF</th>
<th align="left">ExAC_AF</th>
<th align="left">ExAC_Adj_AF</th>
<th align="left">ExAC_AFR_AF</th>
<th align="left">ExAC_AMR_AF</th>
<th align="left">ExAC_EAS_AF</th>
<th align="left">ExAC_FIN_AF</th>
<th align="left">ExAC_NFE_AF</th>
<th align="left">ExAC_OTH_AF</th>
<th align="left">ExAC_SAS_AF</th>
<th align="left">MAX_AF</th>
<th align="left">MAX_AF_POPS</th>
<th align="left">CLIN_SIG</th>
<th align="left">SOMATIC</th>
<th align="left">PHENO</th>
<th align="left">PUBMED</th>
<th align="left">MOTIF_NAME</th>
<th align="left">MOTIF_POS</th>
<th align="left">HIGH_INF_POS</th>
<th align="left">MOTIF_SCORE_CHANGE</th>
<th align="left">IMPACT_rank</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">chr20_30948281</td>
<td align="left">mutect</td>
<td align="left">chr20</td>
<td align="right">30948281</td>
<td align="right">30948281</td>
<td align="left">A</td>
<td align="left">G</td>
<td align="right">11.000</td>
<td align="left">PASS</td>
<td align="left">0/1</td>
<td align="right">3</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">3</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">3</td>
<td align="right">30948281</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">66</td>
<td align="left">Sample1</td>
<td align="left">G</td>
<td align="left">intron_variant</td>
<td align="left">MODIFIER</td>
<td align="left">ASXL1</td>
<td align="left">ENSG00000171456</td>
<td align="left">Transcript</td>
<td align="left">ENST00000306058</td>
<td align="left">protein_coding</td>
<td align="left"></td>
<td align="left">1/11</td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left">rs2208131</td>
<td align="left"></td>
<td align="left">1</td>
<td align="left"></td>
<td align="left">SNV</td>
<td align="left">HGNC</td>
<td align="left">18318</td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left">ENSP00000305119</td>
<td align="left"></td>
<td align="left">A6NIZ6</td>
<td align="left">UPI00015DF8B6</td>
<td align="left">1</td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left">0.4093</td>
<td align="left">0.1672</td>
<td align="left">0.4337</td>
<td align="left">0.744</td>
<td align="left">0.3499</td>
<td align="left">0.4356</td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left">0.744</td>
<td align="left">EAS</td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left">1</td>
</tr>
<tr class="even">
<td align="left">chr20_30948281</td>
<td align="left">mutect</td>
<td align="left">chr20</td>
<td align="right">30948281</td>
<td align="right">30948281</td>
<td align="left">A</td>
<td align="left">G</td>
<td align="right">11.000</td>
<td align="left">PASS</td>
<td align="left">0/1</td>
<td align="right">3</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">3</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">3</td>
<td align="right">30948281</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">66</td>
<td align="left">Sample1</td>
<td align="left">G</td>
<td align="left">intron_variant</td>
<td align="left">MODIFIER</td>
<td align="left">ASXL1</td>
<td align="left">ENSG00000171456</td>
<td align="left">Transcript</td>
<td align="left">ENST00000375687</td>
<td align="left">protein_coding</td>
<td align="left"></td>
<td align="left">1/12</td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left">rs2208131</td>
<td align="left"></td>
<td align="left">1</td>
<td align="left"></td>
<td align="left">SNV</td>
<td align="left">HGNC</td>
<td align="left">18318</td>
<td align="left">YES</td>
<td align="left"></td>
<td align="left"></td>
<td align="left">CCDS13201.1</td>
<td align="left">ENSP00000364839</td>
<td align="left">Q8IXJ9</td>
<td align="left"></td>
<td align="left">UPI000036702C</td>
<td align="left">1</td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left">0.4093</td>
<td align="left">0.1672</td>
<td align="left">0.4337</td>
<td align="left">0.744</td>
<td align="left">0.3499</td>
<td align="left">0.4356</td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left">0.744</td>
<td align="left">EAS</td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left">1</td>
</tr>
<tr class="odd">
<td align="left">chr20_30948281</td>
<td align="left">mutect</td>
<td align="left">chr20</td>
<td align="right">30948281</td>
<td align="right">30948281</td>
<td align="left">A</td>
<td align="left">G</td>
<td align="right">11.000</td>
<td align="left">PASS</td>
<td align="left">0/1</td>
<td align="right">3</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">3</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">3</td>
<td align="right">30948281</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">66</td>
<td align="left">Sample1</td>
<td align="left">G</td>
<td align="left">intron_variant</td>
<td align="left">MODIFIER</td>
<td align="left">ASXL1</td>
<td align="left">ENSG00000171456</td>
<td align="left">Transcript</td>
<td align="left">ENST00000375689</td>
<td align="left">protein_coding</td>
<td align="left"></td>
<td align="left">1/4</td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left">rs2208131</td>
<td align="left"></td>
<td align="left">1</td>
<td align="left"></td>
<td align="left">SNV</td>
<td align="left">HGNC</td>
<td align="left">18318</td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left">ENSP00000364841</td>
<td align="left"></td>
<td align="left">Q5JWS8</td>
<td align="left">UPI00004A2D4D</td>
<td align="left">1</td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left">0.4093</td>
<td align="left">0.1672</td>
<td align="left">0.4337</td>
<td align="left">0.744</td>
<td align="left">0.3499</td>
<td align="left">0.4356</td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left">0.744</td>
<td align="left">EAS</td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left">1</td>
</tr>
<tr class="even">
<td align="left">chr20_30948281</td>
<td align="left">mutect</td>
<td align="left">chr20</td>
<td align="right">30948281</td>
<td align="right">30948281</td>
<td align="left">A</td>
<td align="left">G</td>
<td align="right">11.000</td>
<td align="left">PASS</td>
<td align="left">0/1</td>
<td align="right">3</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">3</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">3</td>
<td align="right">30948281</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">66</td>
<td align="left">Sample1</td>
<td align="left">G</td>
<td align="left">intron_variant</td>
<td align="left">MODIFIER</td>
<td align="left">ASXL1</td>
<td align="left">ENSG00000171456</td>
<td align="left">Transcript</td>
<td align="left">ENST00000497249</td>
<td align="left">protein_coding</td>
<td align="left"></td>
<td align="left">1/4</td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left">rs2208131</td>
<td align="left"></td>
<td align="left">1</td>
<td align="left">cds_start_NF</td>
<td align="left">SNV</td>
<td align="left">HGNC</td>
<td align="left">18318</td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left">ENSP00000451216</td>
<td align="left"></td>
<td align="left"></td>
<td align="left">UPI00021CEF59</td>
<td align="left">1</td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left">0.4093</td>
<td align="left">0.1672</td>
<td align="left">0.4337</td>
<td align="left">0.744</td>
<td align="left">0.3499</td>
<td align="left">0.4356</td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left">0.744</td>
<td align="left">EAS</td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left">1</td>
</tr>
<tr class="odd">
<td align="left">chr20_30948281</td>
<td align="left">mutect</td>
<td align="left">chr20</td>
<td align="right">30948281</td>
<td align="right">30948281</td>
<td align="left">A</td>
<td align="left">G</td>
<td align="right">11.000</td>
<td align="left">PASS</td>
<td align="left">0/1</td>
<td align="right">3</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">3</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">3</td>
<td align="right">30948281</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">66</td>
<td align="left">Sample1</td>
<td align="left">G</td>
<td align="left">intron_variant</td>
<td align="left">MODIFIER</td>
<td align="left">ASXL1</td>
<td align="left">ENSG00000171456</td>
<td align="left">Transcript</td>
<td align="left">ENST00000542461</td>
<td align="left">protein_coding</td>
<td align="left"></td>
<td align="left">1/5</td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left">rs2208131</td>
<td align="left"></td>
<td align="left">1</td>
<td align="left"></td>
<td align="left">SNV</td>
<td align="left">HGNC</td>
<td align="left">18318</td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left">ENSP00000438654</td>
<td align="left"></td>
<td align="left">Q6P1M8</td>
<td align="left">UPI0000232A47</td>
<td align="left">1</td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left">0.4093</td>
<td align="left">0.1672</td>
<td align="left">0.4337</td>
<td align="left">0.744</td>
<td align="left">0.3499</td>
<td align="left">0.4356</td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left">0.744</td>
<td align="left">EAS</td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left">1</td>
</tr>
<tr class="even">
<td align="left">chr20_30948281</td>
<td align="left">mutect</td>
<td align="left">chr20</td>
<td align="right">30948281</td>
<td align="right">30948281</td>
<td align="left">A</td>
<td align="left">G</td>
<td align="right">11.000</td>
<td align="left">PASS</td>
<td align="left">0/1</td>
<td align="right">3</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">3</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">3</td>
<td align="right">30948281</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">66</td>
<td align="left">Sample1</td>
<td align="left">G</td>
<td align="left">non_coding_transcript_exon_variant</td>
<td align="left">MODIFIER</td>
<td align="left">ASXL1</td>
<td align="left">ENSG00000171456</td>
<td align="left">Transcript</td>
<td align="left">ENST00000555343</td>
<td align="left">processed_transcript</td>
<td align="left">1/7</td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left">462</td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left">rs2208131</td>
<td align="left"></td>
<td align="left">1</td>
<td align="left"></td>
<td align="left">SNV</td>
<td align="left">HGNC</td>
<td align="left">18318</td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left">1</td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left">0.4093</td>
<td align="left">0.1672</td>
<td align="left">0.4337</td>
<td align="left">0.744</td>
<td align="left">0.3499</td>
<td align="left">0.4356</td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left">0.744</td>
<td align="left">EAS</td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left">1</td>
</tr>
<tr class="odd">
<td align="left">chr20_30953162</td>
<td align="left">mutect</td>
<td align="left">chr20</td>
<td align="right">30953162</td>
<td align="right">30953162</td>
<td align="left">T</td>
<td align="left">C</td>
<td align="right">15.125</td>
<td align="left">PASS</td>
<td align="left">0/1</td>
<td align="right">4</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">4</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">4</td>
<td align="right">0</td>
<td align="right">30953162</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">121</td>
<td align="left">Sample1</td>
<td align="left">C</td>
<td align="left">intron_variant</td>
<td align="left">MODIFIER</td>
<td align="left">ASXL1</td>
<td align="left">ENSG00000171456</td>
<td align="left">Transcript</td>
<td align="left">ENST00000306058</td>
<td align="left">protein_coding</td>
<td align="left"></td>
<td align="left">1/11</td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left">rs6141294</td>
<td align="left"></td>
<td align="left">1</td>
<td align="left"></td>
<td align="left">SNV</td>
<td align="left">HGNC</td>
<td align="left">18318</td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left">ENSP00000305119</td>
<td align="left"></td>
<td align="left">A6NIZ6</td>
<td align="left">UPI00015DF8B6</td>
<td align="left">1</td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left">0.4087</td>
<td align="left">0.1672</td>
<td align="left">0.4337</td>
<td align="left">0.744</td>
<td align="left">0.3499</td>
<td align="left">0.4325</td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left">0.744</td>
<td align="left">EAS</td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left">1</td>
</tr>
<tr class="even">
<td align="left">chr20_30953162</td>
<td align="left">mutect</td>
<td align="left">chr20</td>
<td align="right">30953162</td>
<td align="right">30953162</td>
<td align="left">T</td>
<td align="left">C</td>
<td align="right">15.125</td>
<td align="left">PASS</td>
<td align="left">0/1</td>
<td align="right">4</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">4</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">4</td>
<td align="right">0</td>
<td align="right">30953162</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">121</td>
<td align="left">Sample1</td>
<td align="left">C</td>
<td align="left">intron_variant</td>
<td align="left">MODIFIER</td>
<td align="left">ASXL1</td>
<td align="left">ENSG00000171456</td>
<td align="left">Transcript</td>
<td align="left">ENST00000375687</td>
<td align="left">protein_coding</td>
<td align="left"></td>
<td align="left">1/12</td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left">rs6141294</td>
<td align="left"></td>
<td align="left">1</td>
<td align="left"></td>
<td align="left">SNV</td>
<td align="left">HGNC</td>
<td align="left">18318</td>
<td align="left">YES</td>
<td align="left"></td>
<td align="left"></td>
<td align="left">CCDS13201.1</td>
<td align="left">ENSP00000364839</td>
<td align="left">Q8IXJ9</td>
<td align="left"></td>
<td align="left">UPI000036702C</td>
<td align="left">1</td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left">0.4087</td>
<td align="left">0.1672</td>
<td align="left">0.4337</td>
<td align="left">0.744</td>
<td align="left">0.3499</td>
<td align="left">0.4325</td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left">0.744</td>
<td align="left">EAS</td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left">1</td>
</tr>
<tr class="odd">
<td align="left">chr20_30953162</td>
<td align="left">mutect</td>
<td align="left">chr20</td>
<td align="right">30953162</td>
<td align="right">30953162</td>
<td align="left">T</td>
<td align="left">C</td>
<td align="right">15.125</td>
<td align="left">PASS</td>
<td align="left">0/1</td>
<td align="right">4</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">4</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">4</td>
<td align="right">0</td>
<td align="right">30953162</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">121</td>
<td align="left">Sample1</td>
<td align="left">C</td>
<td align="left">intron_variant</td>
<td align="left">MODIFIER</td>
<td align="left">ASXL1</td>
<td align="left">ENSG00000171456</td>
<td align="left">Transcript</td>
<td align="left">ENST00000375689</td>
<td align="left">protein_coding</td>
<td align="left"></td>
<td align="left">1/4</td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left">rs6141294</td>
<td align="left"></td>
<td align="left">1</td>
<td align="left"></td>
<td align="left">SNV</td>
<td align="left">HGNC</td>
<td align="left">18318</td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left">ENSP00000364841</td>
<td align="left"></td>
<td align="left">Q5JWS8</td>
<td align="left">UPI00004A2D4D</td>
<td align="left">1</td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left">0.4087</td>
<td align="left">0.1672</td>
<td align="left">0.4337</td>
<td align="left">0.744</td>
<td align="left">0.3499</td>
<td align="left">0.4325</td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left">0.744</td>
<td align="left">EAS</td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left">1</td>
</tr>
<tr class="even">
<td align="left">chr20_30953162</td>
<td align="left">mutect</td>
<td align="left">chr20</td>
<td align="right">30953162</td>
<td align="right">30953162</td>
<td align="left">T</td>
<td align="left">C</td>
<td align="right">15.125</td>
<td align="left">PASS</td>
<td align="left">0/1</td>
<td align="right">4</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">4</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">4</td>
<td align="right">0</td>
<td align="right">30953162</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">121</td>
<td align="left">Sample1</td>
<td align="left">C</td>
<td align="left">upstream_gene_variant</td>
<td align="left">MODIFIER</td>
<td align="left">ASXL1</td>
<td align="left">ENSG00000171456</td>
<td align="left">Transcript</td>
<td align="left">ENST00000470145</td>
<td align="left">processed_transcript</td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left">rs6141294</td>
<td align="left">1025</td>
<td align="left">1</td>
<td align="left"></td>
<td align="left">SNV</td>
<td align="left">HGNC</td>
<td align="left">18318</td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left">1</td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left">0.4087</td>
<td align="left">0.1672</td>
<td align="left">0.4337</td>
<td align="left">0.744</td>
<td align="left">0.3499</td>
<td align="left">0.4325</td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left">0.744</td>
<td align="left">EAS</td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left"></td>
<td align="left">1</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="combine_and_filter-joint-filtering-and-improved-visualisation-over-time" class="section level1">
<h1 class="hasAnchor">
<a href="#combine_and_filter-joint-filtering-and-improved-visualisation-over-time" class="anchor"></a><code><a href="../reference/combine_and_filter.html">combine_and_filter()</a></code>: joint filtering and improved visualisation over time</h1>
<p>This section is more focused on specific tasks that involve preparing the data for plotting changes over time (usually changes in the VAF). This was a recurring task within the clinical trial structures that led to the development of <code>varikondo</code>. This includes a not too large cohort of patients (<code>PID</code>) followed up over <code>Time</code>, usually before and after treatment, and for which some clinical information, like the percentage of tumour content, are collected over the course of the treatment. Patients usually have an overall <code>Outcome</code> reported, depending on whether they respond to treatment (e.g. Responders, Relapse, Refractory). A more detailed example is described in <a href="https://annaquaglieri16.github.io/varikondo/articles/rna-seq-pipe.html"><code>Example: variant calling in RNA-Seq</code></a>.</p>
<p><code>parse_vcf_ouput()</code> imports one VCF file at a time. The function only support VCF input files which derive from calls done on one single sample. To analyse variants over time across multiple samples (within a patient or cohort-wise), one needs to filter and combine multiple variants files to create a combined dataset to be summarise or visualised. Some variant callers like superFreq or MultiSNV <span class="citation">(Josephidou, Lynch, and Tavaré 2015)</span> are built with this purpose. However, these programs might not be suitable in specific scenarios, like calling complex INDELs in RNA-Seq.</p>
<p><code><a href="../reference/combine_and_filter.html">combine_and_filter()</a></code> helps with this. <code><a href="../reference/combine_and_filter.html">combine_and_filter()</a></code> takes in input the combined data frame of variants imported with <code><a href="../reference/parse_vcf_output.html">parse_vcf_output()</a></code> and stacked together with functions like <code><a href="https://www.rdocumentation.org/packages/base/topics/cbind">rbind()</a></code> or <code><a href="https://dplyr.tidyverse.org/reference/bind.html">bind_rows()</a></code>; a <code>patientID</code>; a set of <code>studyGenes</code> if available; and some quality threshold to use for filtering. The output will be a <em>filtered</em> and <em>filled</em> data frame.</p>
<ul>
<li><p><em>Filtered</em>. Similarly to <code><a href="../reference/import_goi_superfreq.html">import_goi_superfreq()</a></code>, variants are kept of only if they pass the defined thresholds at one time point within a patient.</p></li>
<li><p><em>Filled</em>. If a variant is called at one time point but not found at others, <code><a href="../reference/combine_and_filter.html">combine_and_filter()</a></code> imputes <code>0</code> default values for VAF, alt_depth and ref_depth. This feature was added to allow that all patient’s time points available in the <code>clinicalData</code> (metadata) are represented for every variant. This allows a complete visual assessment of a variant over time. If this is not of interest to the user, variants with <code>VAF == 0</code> can be filtered out.</p></li>
</ul>
<p>Below is an illustrative example with sample data, adapted from the type of clinical trials that we usually work on.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1">sample_variants &lt;-<span class="st"> </span>example_variants</a>
<a class="sourceLine" id="cb8-2" data-line-number="2">knitr<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/knitr/topics/kable">kable</a></span>(example_variants)</a></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left">SampleName</th>
<th align="left">PID</th>
<th align="left">Time</th>
<th align="left">mutation_key</th>
<th align="left">chrom</th>
<th align="left">pos</th>
<th align="left">alt</th>
<th align="left">ref</th>
<th align="right">ref_depth</th>
<th align="right">VAF</th>
<th align="left">mutation_det</th>
<th align="left">SYMBOL</th>
<th align="left">Consequence</th>
<th align="right">alt_depth</th>
<th align="left">variant_type</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">P1.Screen</td>
<td align="left">P1</td>
<td align="left">Screen</td>
<td align="left">Gene1-chr1_10-A-ACT</td>
<td align="left">chr1</td>
<td align="left">10</td>
<td align="left">ACT</td>
<td align="left">A</td>
<td align="right">12</td>
<td align="right">0.55</td>
<td align="left">Gene1-Nonsyn</td>
<td align="left">Gene1</td>
<td align="left">Nonsyn</td>
<td align="right">7</td>
<td align="left">INDEL</td>
</tr>
<tr class="even">
<td align="left">P1.Screen</td>
<td align="left">P1</td>
<td align="left">Screen</td>
<td align="left">Gene2-chr1_20-A-ACGTCG</td>
<td align="left">chr1</td>
<td align="left">20</td>
<td align="left">ACGTCG</td>
<td align="left">A</td>
<td align="right">11</td>
<td align="right">0.40</td>
<td align="left">Gene2-Nonsyn</td>
<td align="left">Gene2</td>
<td align="left">Nonsyn</td>
<td align="right">4</td>
<td align="left">INDEL</td>
</tr>
<tr class="odd">
<td align="left">P1.Screen</td>
<td align="left">P1</td>
<td align="left">Screen</td>
<td align="left">Gene3-chr2_30-A-AGG</td>
<td align="left">chr2</td>
<td align="left">30</td>
<td align="left">AGG</td>
<td align="left">A</td>
<td align="right">9</td>
<td align="right">0.20</td>
<td align="left">Gene3-Damaging</td>
<td align="left">Gene3</td>
<td align="left">Damaging</td>
<td align="right">2</td>
<td align="left">INDEL</td>
</tr>
<tr class="even">
<td align="left">P1.Cyc1</td>
<td align="left">P1</td>
<td align="left">Cyc1</td>
<td align="left">Gene1-chr1_10-A-ACT</td>
<td align="left">chr1</td>
<td align="left">10</td>
<td align="left">ACT</td>
<td align="left">A</td>
<td align="right">9</td>
<td align="right">0.48</td>
<td align="left">Gene1-Nonsyn</td>
<td align="left">Gene1</td>
<td align="left">Nonsyn</td>
<td align="right">4</td>
<td align="left">INDEL</td>
</tr>
<tr class="odd">
<td align="left">P1.Cyc1</td>
<td align="left">P1</td>
<td align="left">Cyc1</td>
<td align="left">Gene2-chr1_20-A-ACGTCG</td>
<td align="left">chr1</td>
<td align="left">20</td>
<td align="left">ACGTCG</td>
<td align="left">A</td>
<td align="right">12</td>
<td align="right">0.45</td>
<td align="left">Gene2-Nonsyn</td>
<td align="left">Gene2</td>
<td align="left">Nonsyn</td>
<td align="right">5</td>
<td align="left">INDEL</td>
</tr>
<tr class="even">
<td align="left">P1.Cyc1</td>
<td align="left">P1</td>
<td align="left">Cyc1</td>
<td align="left">Gene3-chr2_30-A-AGG</td>
<td align="left">chr2</td>
<td align="left">30</td>
<td align="left">AGG</td>
<td align="left">A</td>
<td align="right">8</td>
<td align="right">0.25</td>
<td align="left">Gene3-Damaging</td>
<td align="left">Gene3</td>
<td align="left">Damaging</td>
<td align="right">2</td>
<td align="left">INDEL</td>
</tr>
<tr class="odd">
<td align="left">P1.Cyc2</td>
<td align="left">P1</td>
<td align="left">Cyc2</td>
<td align="left">Gene1-chr1_10-A-ACT</td>
<td align="left">chr1</td>
<td align="left">10</td>
<td align="left">ACT</td>
<td align="left">A</td>
<td align="right">13</td>
<td align="right">0.45</td>
<td align="left">Gene1-Nonsyn</td>
<td align="left">Gene1</td>
<td align="left">Nonsyn</td>
<td align="right">6</td>
<td align="left">INDEL</td>
</tr>
<tr class="even">
<td align="left">P1.Cyc2</td>
<td align="left">P1</td>
<td align="left">Cyc2</td>
<td align="left">Gene2-chr1_20-A-ACGTCG</td>
<td align="left">chr1</td>
<td align="left">20</td>
<td align="left">ACGTCG</td>
<td align="left">A</td>
<td align="right">14</td>
<td align="right">0.50</td>
<td align="left">Gene2-Nonsyn</td>
<td align="left">Gene2</td>
<td align="left">Nonsyn</td>
<td align="right">7</td>
<td align="left">INDEL</td>
</tr>
<tr class="odd">
<td align="left">P2.Screen</td>
<td align="left">P2</td>
<td align="left">Screen</td>
<td align="left">Gene1-chr1_10-G-A</td>
<td align="left">chr1</td>
<td align="left">10</td>
<td align="left">A</td>
<td align="left">G</td>
<td align="right">20</td>
<td align="right">0.50</td>
<td align="left">Gene1-Nonsyn</td>
<td align="left">Gene1</td>
<td align="left">Nonsyn</td>
<td align="right">10</td>
<td align="left">SNV</td>
</tr>
<tr class="even">
<td align="left">P2.Cyc1</td>
<td align="left">P2</td>
<td align="left">Cyc1</td>
<td align="left">Gene1-chr1_10-G-A</td>
<td align="left">chr1</td>
<td align="left">10</td>
<td align="left">A</td>
<td align="left">G</td>
<td align="right">25</td>
<td align="right">0.55</td>
<td align="left">Gene1-Nonsyn</td>
<td align="left">Gene1</td>
<td align="left">Nonsyn</td>
<td align="right">14</td>
<td align="left">SNV</td>
</tr>
<tr class="odd">
<td align="left">P2.Cyc2</td>
<td align="left">P2</td>
<td align="left">Cyc2</td>
<td align="left">Gene1-chr1_10-G-A</td>
<td align="left">chr1</td>
<td align="left">10</td>
<td align="left">A</td>
<td align="left">G</td>
<td align="right">23</td>
<td align="right">0.45</td>
<td align="left">Gene1-Nonsyn</td>
<td align="left">Gene1</td>
<td align="left">Nonsyn</td>
<td align="right">10</td>
<td align="left">SNV</td>
</tr>
<tr class="even">
<td align="left">P2.Cyc2</td>
<td align="left">P2</td>
<td align="left">Cyc2</td>
<td align="left">Gene4-chr4_40-T-C</td>
<td align="left">chr4</td>
<td align="left">40</td>
<td align="left">C</td>
<td align="left">T</td>
<td align="right">15</td>
<td align="right">0.10</td>
<td align="left">Gene4-Damaging</td>
<td align="left">Gene4</td>
<td align="left">Damaging</td>
<td align="right">2</td>
<td align="left">SNV</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1">sample_metadata &lt;-<span class="st"> </span>example_metadata</a>
<a class="sourceLine" id="cb9-2" data-line-number="2">knitr<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/knitr/topics/kable">kable</a></span>(example_metadata)</a></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left">SampleName</th>
<th align="left">PID</th>
<th align="left">Time</th>
<th align="right">Blast</th>
<th align="left">Outcome</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">P1.Screen</td>
<td align="left">P1</td>
<td align="left">Screen</td>
<td align="right">0.78</td>
<td align="left">Relapse</td>
</tr>
<tr class="even">
<td align="left">P1.Rem</td>
<td align="left">P1</td>
<td align="left">Rem</td>
<td align="right">0.05</td>
<td align="left">Relapse</td>
</tr>
<tr class="odd">
<td align="left">P1.Cyc1</td>
<td align="left">P1</td>
<td align="left">Cyc1</td>
<td align="right">0.10</td>
<td align="left">Relapse</td>
</tr>
<tr class="even">
<td align="left">P1.Cyc2</td>
<td align="left">P1</td>
<td align="left">Cyc2</td>
<td align="right">0.50</td>
<td align="left">Relapse</td>
</tr>
<tr class="odd">
<td align="left">P2.Screen</td>
<td align="left">P2</td>
<td align="left">Screen</td>
<td align="right">0.90</td>
<td align="left">Relapse</td>
</tr>
<tr class="even">
<td align="left">P2.Rem</td>
<td align="left">P2</td>
<td align="left">Rem</td>
<td align="right">0.00</td>
<td align="left">Relapse</td>
</tr>
<tr class="odd">
<td align="left">P2.Cyc1</td>
<td align="left">P2</td>
<td align="left">Cyc1</td>
<td align="right">0.05</td>
<td align="left">Relapse</td>
</tr>
<tr class="even">
<td align="left">P2.Cyc2</td>
<td align="left">P2</td>
<td align="left">Cyc2</td>
<td align="right">0.20</td>
<td align="left">Relapse</td>
</tr>
</tbody>
</table>
<p><code><a href="../reference/import_clinical.html">import_clinical()</a></code> was developed with the mere function to reshape clinical data, like % blast content ( = tumour content), into the same format as we store variants information to make it easier to be combined and plotted together with the VAF over time.</p>
<p>This can be useful to compare patient’s clinical assessment with the cancer molecular response to treatment. For example, there can be cases where even though the patient does not seem to respond to treatment (tumour content does not drop after treatment), some cancer variants do individually respond.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb10-1" data-line-number="1">combine_and_fill &lt;-<span class="st"> </span>varikondo<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/varikondo/topics/combine_and_filter">combine_and_filter</a></span>(<span class="dt">variants =</span> sample_variants,</a>
<a class="sourceLine" id="cb10-2" data-line-number="2">                               <span class="dt">patientID =</span> <span class="st">"P1"</span>,</a>
<a class="sourceLine" id="cb10-3" data-line-number="3">                              <span class="co"># studyGenes = c("Gene1","Gene2","Gene3"),</span></a>
<a class="sourceLine" id="cb10-4" data-line-number="4">                               <span class="dt">clinicalData =</span> sample_metadata,</a>
<a class="sourceLine" id="cb10-5" data-line-number="5">                               <span class="dt">min_vaf =</span> <span class="fl">0.05</span>,</a>
<a class="sourceLine" id="cb10-6" data-line-number="6">                               <span class="dt">min_alt =</span> <span class="dv">2</span>)</a>
<a class="sourceLine" id="cb10-7" data-line-number="7"><span class="co">#&gt; IMPACT is missing and will be filled with NAs.</span></a>
<a class="sourceLine" id="cb10-8" data-line-number="8"><span class="co">#&gt; qual is missing and will be filled with 0 and minQual = 0.</span></a>
<a class="sourceLine" id="cb10-9" data-line-number="9"></a>
<a class="sourceLine" id="cb10-10" data-line-number="10">knitr<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/knitr/topics/kable">kable</a></span>(combine_and_fill,<span class="dt">caption =</span> <span class="st">"Output from combine_and_filter() for patient P1 in example_variants."</span>)</a></code></pre></div>
<table class="table">
<caption>Output from combine_and_filter() for patient P1 in example_variants.</caption>
<thead><tr class="header">
<th align="left">SampleName</th>
<th align="left">PID</th>
<th align="left">Time</th>
<th align="left">chrom</th>
<th align="left">pos</th>
<th align="left">ref</th>
<th align="left">alt</th>
<th align="left">Location</th>
<th align="left">mutation_det</th>
<th align="left">mutation_key</th>
<th align="left">SYMBOL</th>
<th align="left">Consequence</th>
<th align="left">variant_type</th>
<th align="right">Blast</th>
<th align="left">Outcome</th>
<th align="right">ref_depth</th>
<th align="right">VAF</th>
<th align="right">alt_depth</th>
<th align="left">IMPACT</th>
<th align="right">qual</th>
<th align="right">tot_depth</th>
<th align="right">IMPACT_rank</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">P1.Cyc1</td>
<td align="left">P1</td>
<td align="left">Cyc1</td>
<td align="left">chr1</td>
<td align="left">20</td>
<td align="left">A</td>
<td align="left">ACGTCG</td>
<td align="left">chr1_20</td>
<td align="left">Gene2 Nonsyn</td>
<td align="left">chr1-20-A-ACGTCG</td>
<td align="left">Gene2</td>
<td align="left">Nonsyn</td>
<td align="left">INDEL</td>
<td align="right">0.10</td>
<td align="left">Relapse</td>
<td align="right">12</td>
<td align="right">0.2941176</td>
<td align="right">5</td>
<td align="left">NA</td>
<td align="right">0</td>
<td align="right">17</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">P1.Cyc2</td>
<td align="left">P1</td>
<td align="left">Cyc2</td>
<td align="left">chr1</td>
<td align="left">10</td>
<td align="left">A</td>
<td align="left">ACT</td>
<td align="left">chr1_10</td>
<td align="left">Gene1 Nonsyn</td>
<td align="left">chr1-10-A-ACT</td>
<td align="left">Gene1</td>
<td align="left">Nonsyn</td>
<td align="left">INDEL</td>
<td align="right">0.50</td>
<td align="left">Relapse</td>
<td align="right">13</td>
<td align="right">0.3157895</td>
<td align="right">6</td>
<td align="left">NA</td>
<td align="right">0</td>
<td align="right">19</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">P1.Cyc2</td>
<td align="left">P1</td>
<td align="left">Cyc2</td>
<td align="left">chr1</td>
<td align="left">20</td>
<td align="left">A</td>
<td align="left">ACGTCG</td>
<td align="left">chr1_20</td>
<td align="left">Gene2 Nonsyn</td>
<td align="left">chr1-20-A-ACGTCG</td>
<td align="left">Gene2</td>
<td align="left">Nonsyn</td>
<td align="left">INDEL</td>
<td align="right">0.50</td>
<td align="left">Relapse</td>
<td align="right">14</td>
<td align="right">0.3333333</td>
<td align="right">7</td>
<td align="left">NA</td>
<td align="right">0</td>
<td align="right">21</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">P1.Screen</td>
<td align="left">P1</td>
<td align="left">Screen</td>
<td align="left">chr1</td>
<td align="left">10</td>
<td align="left">A</td>
<td align="left">ACT</td>
<td align="left">chr1_10</td>
<td align="left">Gene1 Nonsyn</td>
<td align="left">chr1-10-A-ACT</td>
<td align="left">Gene1</td>
<td align="left">Nonsyn</td>
<td align="left">INDEL</td>
<td align="right">0.78</td>
<td align="left">Relapse</td>
<td align="right">12</td>
<td align="right">0.3684211</td>
<td align="right">7</td>
<td align="left">NA</td>
<td align="right">0</td>
<td align="right">19</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">P1.Screen</td>
<td align="left">P1</td>
<td align="left">Screen</td>
<td align="left">chr1</td>
<td align="left">20</td>
<td align="left">A</td>
<td align="left">ACGTCG</td>
<td align="left">chr1_20</td>
<td align="left">Gene2 Nonsyn</td>
<td align="left">chr1-20-A-ACGTCG</td>
<td align="left">Gene2</td>
<td align="left">Nonsyn</td>
<td align="left">INDEL</td>
<td align="right">0.78</td>
<td align="left">Relapse</td>
<td align="right">11</td>
<td align="right">0.2666667</td>
<td align="right">4</td>
<td align="left">NA</td>
<td align="right">0</td>
<td align="right">15</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">P1.Cyc1</td>
<td align="left">P1</td>
<td align="left">Cyc1</td>
<td align="left">chr1</td>
<td align="left">10</td>
<td align="left">A</td>
<td align="left">ACT</td>
<td align="left">chr1_10</td>
<td align="left">Gene1 Nonsyn</td>
<td align="left">chr1-10-A-ACT</td>
<td align="left">Gene1</td>
<td align="left">Nonsyn</td>
<td align="left">INDEL</td>
<td align="right">0.10</td>
<td align="left">Relapse</td>
<td align="right">9</td>
<td align="right">0.3076923</td>
<td align="right">4</td>
<td align="left">NA</td>
<td align="right">0</td>
<td align="right">13</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">P1.Rem</td>
<td align="left">P1</td>
<td align="left">Rem</td>
<td align="left">chr1</td>
<td align="left">10</td>
<td align="left">A</td>
<td align="left">ACT</td>
<td align="left">chr1_10</td>
<td align="left">Gene1 Nonsyn</td>
<td align="left">chr1-10-A-ACT</td>
<td align="left">Gene1</td>
<td align="left">Nonsyn</td>
<td align="left">INDEL</td>
<td align="right">0.05</td>
<td align="left">Relapse</td>
<td align="right">0</td>
<td align="right">0.0000000</td>
<td align="right">0</td>
<td align="left">NA</td>
<td align="right">NA</td>
<td align="right">0</td>
<td align="right">NA</td>
</tr>
<tr class="even">
<td align="left">P1.Rem</td>
<td align="left">P1</td>
<td align="left">Rem</td>
<td align="left">chr1</td>
<td align="left">20</td>
<td align="left">A</td>
<td align="left">ACGTCG</td>
<td align="left">chr1_20</td>
<td align="left">Gene2 Nonsyn</td>
<td align="left">chr1-20-A-ACGTCG</td>
<td align="left">Gene2</td>
<td align="left">Nonsyn</td>
<td align="left">INDEL</td>
<td align="right">0.05</td>
<td align="left">Relapse</td>
<td align="right">0</td>
<td align="right">0.0000000</td>
<td align="right">0</td>
<td align="left">NA</td>
<td align="right">NA</td>
<td align="right">0</td>
<td align="right">NA</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" data-line-number="1">import_blast &lt;-<span class="st"> </span>varikondo<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/varikondo/topics/import_clinical">import_clinical</a></span>(<span class="dt">clinicalData =</span> sample_metadata,</a>
<a class="sourceLine" id="cb11-2" data-line-number="2">                                <span class="dt">patientID =</span> <span class="st">"P1"</span>,</a>
<a class="sourceLine" id="cb11-3" data-line-number="3">                                <span class="dt">extract_column =</span> <span class="st">"Blast"</span>) </a>
<a class="sourceLine" id="cb11-4" data-line-number="4">knitr<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/knitr/topics/kable">kable</a></span>(import_blast,<span class="dt">caption =</span> <span class="st">"Output from import_clinical() for patient P1 in example_metadata."</span>)</a></code></pre></div>
<table class="table">
<caption>Output from import_clinical() for patient P1 in example_metadata.</caption>
<thead><tr class="header">
<th align="left">SampleName</th>
<th align="left">PID</th>
<th align="left">Time</th>
<th align="right">VAF</th>
<th align="left">mutation_key</th>
<th align="left">mutation_det</th>
<th align="left">variant_type</th>
<th align="left">SYMBOL</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">P1.Screen</td>
<td align="left">P1</td>
<td align="left">Screen</td>
<td align="right">0.78</td>
<td align="left">Blast</td>
<td align="left">Blast</td>
<td align="left">Blast</td>
<td align="left">Blast</td>
</tr>
<tr class="even">
<td align="left">P1.Rem</td>
<td align="left">P1</td>
<td align="left">Rem</td>
<td align="right">0.05</td>
<td align="left">Blast</td>
<td align="left">Blast</td>
<td align="left">Blast</td>
<td align="left">Blast</td>
</tr>
<tr class="odd">
<td align="left">P1.Cyc1</td>
<td align="left">P1</td>
<td align="left">Cyc1</td>
<td align="right">0.10</td>
<td align="left">Blast</td>
<td align="left">Blast</td>
<td align="left">Blast</td>
<td align="left">Blast</td>
</tr>
<tr class="even">
<td align="left">P1.Cyc2</td>
<td align="left">P1</td>
<td align="left">Cyc2</td>
<td align="right">0.50</td>
<td align="left">Blast</td>
<td align="left">Blast</td>
<td align="left">Blast</td>
<td align="left">Blast</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb12-1" data-line-number="1">variants &lt;-<span class="st"> </span>example_variants <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb12-2" data-line-number="2"><span class="st">  </span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/bind.html">bind_rows</a></span>(import_blast)</a>
<a class="sourceLine" id="cb12-3" data-line-number="3"></a>
<a class="sourceLine" id="cb12-4" data-line-number="4">p1 =<span class="st"> </span>variants <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb12-5" data-line-number="5"><span class="st">  </span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span>(PID <span class="op">%in%</span><span class="st"> "P1"</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb12-6" data-line-number="6"><span class="st"> </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span>(<span class="kw"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="dt">x =</span> Time, <span class="dt">y =</span> VAF,<span class="dt">colour =</span> mutation_det,<span class="dt">group =</span> mutation_key)) <span class="op">+</span></a>
<a class="sourceLine" id="cb12-7" data-line-number="7"><span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html">geom_point</a></span>() <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb12-8" data-line-number="8"><span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/geom_path.html">geom_line</a></span>() <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb12-9" data-line-number="9"><span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_bw</a></span>() <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb12-10" data-line-number="10"><span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span>(<span class="dt">legend.position =</span> <span class="st">"bottom"</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb12-11" data-line-number="11"><span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span>(<span class="st">"P1 - before combine_and_filter()"</span>)</a>
<a class="sourceLine" id="cb12-12" data-line-number="12"></a>
<a class="sourceLine" id="cb12-13" data-line-number="13"></a>
<a class="sourceLine" id="cb12-14" data-line-number="14">combine_and_fill &lt;-<span class="st"> </span>combine_and_fill <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb12-15" data-line-number="15"><span class="st">  </span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/bind.html">bind_rows</a></span>(import_blast)</a>
<a class="sourceLine" id="cb12-16" data-line-number="16"></a>
<a class="sourceLine" id="cb12-17" data-line-number="17">p2 =<span class="st"> </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span>(combine_and_fill,<span class="kw"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="dt">x =</span> Time, <span class="dt">y =</span> VAF,</a>
<a class="sourceLine" id="cb12-18" data-line-number="18">                               <span class="dt">colour=</span>mutation_det,<span class="dt">group=</span>mutation_key)) <span class="op">+</span></a>
<a class="sourceLine" id="cb12-19" data-line-number="19"><span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html">geom_point</a></span>() <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb12-20" data-line-number="20"><span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/geom_path.html">geom_line</a></span>() <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb12-21" data-line-number="21"><span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_bw</a></span>()<span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb12-22" data-line-number="22"><span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span>(<span class="dt">legend.position =</span> <span class="st">"bottom"</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb12-23" data-line-number="23"><span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span>(<span class="st">"P1 - after combine_and_filter()"</span>)</a>
<a class="sourceLine" id="cb12-24" data-line-number="24"></a>
<a class="sourceLine" id="cb12-25" data-line-number="25">legend =<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/cowplot/topics/get_legend">get_legend</a></span>(p1)</a>
<a class="sourceLine" id="cb12-26" data-line-number="26"></a>
<a class="sourceLine" id="cb12-27" data-line-number="27">sec1 &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/cowplot/topics/plot_grid">plot_grid</a></span>(p1 <span class="op">+</span><span class="st"> </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span>(<span class="dt">legend.position =</span> <span class="st">"none"</span>),</a>
<a class="sourceLine" id="cb12-28" data-line-number="28">          p2 <span class="op">+</span><span class="st"> </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span>(<span class="dt">legend.position =</span> <span class="st">"none"</span>))</a>
<a class="sourceLine" id="cb12-29" data-line-number="29"><span class="kw"><a href="https://www.rdocumentation.org/packages/cowplot/topics/plot_grid">plot_grid</a></span>(sec1,</a>
<a class="sourceLine" id="cb12-30" data-line-number="30">          legend,<span class="dt">nrow=</span><span class="dv">2</span>,<span class="dt">rel_heights =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="dv">4</span>,<span class="dv">1</span>))</a></code></pre></div>
<p><img src="vignette_files/figure-html/plot-patient1-1.png" width="700"></p>
</div>
<div id="import-standardised-superfreq-output" class="section level1">
<h1 class="hasAnchor">
<a href="#import-standardised-superfreq-output" class="anchor"></a>Import standardised <code>superFreq</code> output</h1>
<p><code><a href="../reference/import_goi_superfreq.html">import_goi_superfreq()</a></code> was built to import the output produced by <a href="https://github.com/ChristofferFlensburg/superFreq"><code>superFreq</code></a>, normally stored in nested lists within each patient’s <code>Rdata</code> file or in <code>.csv</code> files. <code>superFreq</code> is used to perform clonal tracking of cancer genomes over time and therefore produces temporal output for somatic SNVs, short INDELs, CNVs and clones. A <code>superFreq</code> run (see <a href="https://github.com/ChristofferFlensburg/superFreq">instructions</a>) will create and <code>R</code> folder with results stored in separate folder by patient. <code>import_goi_supefreq(superFreq_R_path, patientID = "P1)</code> will look into this folder to extract all the results for patient <code>P1</code>.</p>
<p>Usually, collaborators might be interested in summarising a particular subset of genes of interest which can be passed to <code>import_goi_supefreq()</code> as a character vector in <code>studyGenes</code>. If no <code>studyGenes</code> are provided, variants on all genes are returned. <code>ref_genome</code> is necessary since <code>superFreq</code> stores variants using a different system of genomic coordinate which needs to be converted back to cDNA locations. Other variant threshold (<code>min_vaf</code> and <code>min_alt</code>) can be used to filter variants. Within a patients, only variants which pass those thresholds at any time point will be kept.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb13-1" data-line-number="1">import_sf &lt;-<span class="st"> </span>varikondo<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/varikondo/topics/import_goi_superfreq">import_goi_superfreq</a></span>(<span class="dt">superFreq_R_path =</span> superFreq_R_path,</a>
<a class="sourceLine" id="cb13-2" data-line-number="2">   <span class="dt">superFreq_meta_path =</span> superFreq_meta_path,</a>
<a class="sourceLine" id="cb13-3" data-line-number="3">  <span class="dt">patientID =</span> patientID,</a>
<a class="sourceLine" id="cb13-4" data-line-number="4">  <span class="dt">studyGenes =</span> <span class="st">"KIT"</span></a>
<a class="sourceLine" id="cb13-5" data-line-number="5">  <span class="dt">ref_genome =</span> <span class="st">"hg38"</span>,</a>
<a class="sourceLine" id="cb13-6" data-line-number="6">  <span class="dt">min_vaf =</span> <span class="fl">0.05</span>,</a>
<a class="sourceLine" id="cb13-7" data-line-number="7">  <span class="dt">min_alt =</span> <span class="dv">3</span>)</a></code></pre></div>
</div>
<div id="explore-variants-interactively" class="section level1">
<h1 class="hasAnchor">
<a href="#explore-variants-interactively" class="anchor"></a>Explore variants interactively</h1>
<p>Now that variants are standardised and combined, it is possible to explore them interactively using the Shiny App <a href="https://shiny.wehi.edu.au/quaglieri.a/shiny-clone/" class="uri">https://shiny.wehi.edu.au/quaglieri.a/shiny-clone/</a>.</p>
<p><img src="../man/figures/01-explore-mutations.png"></p>
<p>Just save the combined dataset into a <code>.csv</code> file and load it into <a href="https://shiny.wehi.edu.au/quaglieri.a/shiny-clone/" class="uri">https://shiny.wehi.edu.au/quaglieri.a/shiny-clone/</a>.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb14-1" data-line-number="1"><span class="kw">write_csv</span>(combine_and_fill, <span class="st">"../inst/combine_and_fill_explore.csv"</span>)</a></code></pre></div>
<p><img src="../man/figures/02-load-variants.png"></p>
<p>Finally, before starting to explore, select the order in which the time variable should be plotted.</p>
<p><img src="../man/figures/03-select-order-time.png"></p>
<p>Start plotting!</p>
<p><img src="../man/figures/04-lineplots-by-patients.png"></p>
</div>
<div id="bibliography" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#bibliography" class="anchor"></a>Bibliography</h1>
<div id="refs" class="references">
<div id="ref-Cibulskis2013-vv">
<p>Cibulskis, Kristian, Michael S Lawrence, Scott L Carter, Andrey Sivachenko, David Jaffe, Carrie Sougnez, Stacey Gabriel, Matthew Meyerson, Eric S Lander, and Gad Getz. 2013. “Sensitive Detection of Somatic Point Mutations in Impure and Heterogeneous Cancer Samples.” <em>Nat. Biotechnol.</em> 31 (3). Nature Research: 213–19.</p>
</div>
<div id="ref-Flensburg2018-ah">
<p>Flensburg, Christoffer, Tobias Sargeant, Alicia Oshlack, and Ian Majewski. 2018. “SuperFreq: Integrated Mutation Detection and Clonal Tracking in Cancer.” <em>bioRxiv</em>.</p>
</div>
<div id="ref-Garrison2012-cl">
<p>Garrison, Erik, and Gabor Marth. 2012. “Haplotype-Based Variant Detection from Short-Read Sequencing,” July.</p>
</div>
<div id="ref-Josephidou2015-cy">
<p>Josephidou, Malvina, Andy G Lynch, and Simon Tavaré. 2015. “multiSNV: A Probabilistic Approach for Improving Detection of Somatic Point Mutations from Multiple Related Tumour Samples.” <em>Nucleic Acids Res.</em> 43 (9): e61.</p>
</div>
<div id="ref-Koboldt2012-wx">
<p>Koboldt, Daniel C, Qunyuan Zhang, David E Larson, Dong Shen, Michael D McLellan, Ling Lin, Christopher A Miller, Elaine R Mardis, Li Ding, and Richard K Wilson. 2012. “VarScan 2: Somatic Mutation and Copy Number Alteration Discovery in Cancer by Exome Sequencing.” <em>Genome Res.</em> 22 (3): 568–76.</p>
</div>
<div id="ref-Lai2016-ws">
<p>Lai, Zhongwu, Aleksandra Markovets, Miika Ahdesmaki, Brad Chapman, Oliver Hofmann, Robert McEwen, Justin Johnson, Brian Dougherty, J Carl Barrett, and Jonathan R Dry. 2016. “VarDict: A Novel and Versatile Variant Caller for Next-Generation Sequencing in Cancer Research.” <em>Nucleic Acids Res.</em> 44 (11): e108.</p>
</div>
<div id="ref-McLaren2016-lv">
<p>McLaren, William, Laurent Gil, Sarah E Hunt, Harpreet Singh Riat, Graham R S Ritchie, Anja Thormann, Paul Flicek, and Fiona Cunningham. 2016. “The Ensembl Variant Effect Predictor.” <em>Genome Biol.</em> 17 (1): 122.</p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#setup">Setup</a></li>
      <li>
<a href="#import-and-parse-a-single-vcf">Import and parse a single <code>VCF</code></a><ul class="nav nav-pills nav-stacked">
<li><a href="#import-only-specific-ranges-of-the-vcf-file">Import only specific ranges of the VCF file</a></li>
      </ul>
</li>
      <li><a href="#combine_and_filter-joint-filtering-and-improved-visualisation-over-time"><code><a href="--/reference/combine_and_filter-html">combine_and_filter()</a></code>: joint filtering and improved visualisation over time</a></li>
      <li><a href="#import-standardised-superfreq-output">Import standardised <code>superFreq</code> output</a></li>
      <li><a href="#explore-variants-interactively">Explore variants interactively</a></li>
      <li><a href="#bibliography">Bibliography</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by <a href="https://github.com/annaquaglieri16">Anna Quaglieri</a>, <a href="https://github.com/ChristofferFlensburg">Christoffer Flensburg</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.9000.</p>
</div>
      </footer>
</div>

  

  </body>
</html>
