<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>General import for variants — import_any • varikondo</title>

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.3.7/sandstone/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script>

<!-- sticky kit -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



<meta property="og:title" content="General import for variants — import_any" />

<meta property="og:description" content="This function will take as input a data frame of variants with specific column information and return a filtered set with sample's clinical infrmation and default variants information also for samples without variants." />
<meta name="twitter:card" content="summary" />



<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">varikondo</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.3.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/vignette.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../articles/standardise-variant-fields.html">Articles</a>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/annaquaglieri16/varikondo">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>General import for variants</h1>
    <small class="dont-index">Source: <a href='https://github.com/annaquaglieri16/varikondo/blob/master/R/import_any.R'><code>R/import_any.R</code></a></small>
    <div class="hidden name"><code>import_any.Rd</code></div>
    </div>

    <div class="ref-description">
    
    <p>This function will take as input a data frame of variants with specific column information and return a filtered set with sample's clinical infrmation and default variants information also for samples without variants.</p>
    
    </div>

    <pre class="usage"><span class='fu'>import_any</span>(<span class='kw'>variants</span> <span class='kw'>=</span> <span class='kw'>NULL</span>, <span class='kw'>patientID</span> <span class='kw'>=</span> <span class='kw'>NULL</span>, <span class='kw'>studyGenes</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>minQual</span> <span class='kw'>=</span> <span class='fl'>20</span>, <span class='kw'>clinicalData</span> <span class='kw'>=</span> <span class='kw'>NULL</span>, <span class='kw'>tidy</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>,
  <span class='kw'>keep_impact</span> <span class='kw'>=</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/c'>c</a></span>(<span class='st'>"HIGH"</span>, <span class='st'>"MODERATE"</span>), <span class='kw'>variant_type</span> <span class='kw'>=</span> <span class='st'>"indels-vardict"</span>)</pre>
    
    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>variants</th>
      <td><p>a data frame where every row is a variant for one sample at a specific time point. The variants can derive from any caller but the input should be standardised to have the following columns: 'SampleName','PID','Time','chrom', 'pos', 'alt', 'ref', 'ref_depth','alt_depth' and (gene) 'SYMBOL' (see more information in Details). The columns `Consequence` and `IMPACT` (as annotated by Variant Effect Predictor (VEP) https://asia.ensembl.org/info/genome/variation/prediction/predicted_data.html) are filled with default values if not found. If VEP `Consequence` is not available it could populated with any other informations like exon number, INDEL/SNV label etc...  to add details to each mutations (useful for plotting purposes). The `SampleName` columns is unique for every sequencing sample while `PID` for every patient.</p></td>
    </tr>
    <tr>
      <th>patientID</th>
      <td><p>a character vector specifying the patient/s id/s for which variants have to be imported.</p></td>
    </tr>
    <tr>
      <th>studyGenes</th>
      <td><p>genes of interest.</p></td>
    </tr>
    <tr>
      <th>minQual</th>
      <td><p>minimum quality for a variant to be kept.</p></td>
    </tr>
    <tr>
      <th>clinicalData</th>
      <td><p>clinical data about the patients in the cohort. It has to contain the a column `SampleName`.</p></td>
    </tr>
    <tr>
      <th>tidy</th>
      <td><p>Logical. Should the ouput be in a tidy or untidy (list of matrices) format? Default is `tidy = TRUE`.</p></td>
    </tr>
    <tr>
      <th>keep_impact</th>
      <td><p>vector specifying the IMPACT values to select variants. Values allowed are HIGH, MODERATE, LOW, MODIFIER ( https://asia.ensembl.org/info/genome/variation/prediction/predicted_data.html). IMPACT should be a columns of `variants`. If it is not found all variants are kept.</p></td>
    </tr>
    <tr>
      <th>variant_type</th>
      <td><p>Label for the type of variants imported, e.g. vardict-indels.</p></td>
    </tr>
    </table>
    
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>This function will keep only the variants for `patientID` found on `studyGenes` and with a `minQual`. If a sample has no variants, then only clinical information will be returned with default values for the variant information.
More details about the `variants` input:</p>
<p>- The `Time` column can be defined in any way and it should reflect the time of sample collection. For example it could be defined as Time0, Time1, Time2 etc...</p>
<p>- If the column `IMPACT` is not found it will be filled with NAs and no variants will be filtered. Otherwise, values of the columns are checked and if they are within the expected values (HIGH, MODERATE, LOW or MODIFIER) only variants with `keep_impact` entries are kept. If a mutation appears twice with different `IMPACT` values only the most damaging will be kept.</p>
<p>The variants are then merged with the clinical information of `patientID`. This step is needed so that if no variants are returned for one time point for one `patientID`, default entries for Variant Allele Frequency (VAF), reference and alterative depths will be created. The default value is 0 for all of the above. A variant is reported for a patient only if at any time point its VAF &gt;= 0.15 and the total depth is &gt;= 10. The function can return the variants in a tidy (long format) or untidy (wide, matrix and list) format. If `tidy = FALSE` the function will return a matrix where the rows are all the unique variants found for `patientID` across time and the columns are the samples of `patientID` across time.</p>
    

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'>
<span class='no'>indels</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/data.frame'>data.frame</a></span>(<span class='kw'>PID</span> <span class='kw'>=</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/rep'>rep</a></span>(<span class='st'>"D1"</span>,<span class='fl'>9</span>),
                     <span class='kw'>Outcome</span> <span class='kw'>=</span> <span class='st'>"Rel"</span>,
                     <span class='kw'>SampleName</span> <span class='kw'>=</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/c'>c</a></span>(<span class='st'>"D1.Screen.Rel"</span>,<span class='st'>"D1.Screen.Rel"</span>,<span class='st'>"D1.Screen.Rel"</span>,
                                    <span class='st'>"D1.Cyc1.Rel"</span>,<span class='st'>"D1.Cyc1.Rel"</span>,<span class='st'>"D1.Cyc1.Rel"</span>,
                                    <span class='st'>"D1.Cyc2.Rel"</span>,<span class='st'>"D1.Cyc2.Rel"</span>,<span class='st'>"D1.Cyc2.Rel"</span>),
                     <span class='kw'>Time</span> <span class='kw'>=</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/c'>c</a></span>(<span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/rep'>rep</a></span>(<span class='st'>"Screen"</span>,<span class='fl'>3</span>),<span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/rep'>rep</a></span>(<span class='st'>"Cyc1"</span>,<span class='fl'>3</span>),<span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/rep'>rep</a></span>(<span class='st'>"Cyc2"</span>,<span class='fl'>3</span>)),
                     <span class='kw'>chrom</span> <span class='kw'>=</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/c'>c</a></span>(<span class='st'>"chr1"</span>,<span class='st'>"chr1"</span>,<span class='st'>"chr2"</span>,<span class='st'>"chr2"</span>,<span class='st'>"chr1"</span>,<span class='st'>"chr1"</span>,<span class='st'>"chr2"</span>,<span class='st'>"chr1"</span>,<span class='st'>"chr2"</span>),
                     <span class='kw'>pos</span> <span class='kw'>=</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/c'>c</a></span>(<span class='fl'>10</span>,<span class='fl'>20</span>,<span class='fl'>30</span>,<span class='fl'>10</span>,<span class='fl'>20</span>,<span class='fl'>30</span>,<span class='fl'>10</span>,<span class='fl'>20</span>,<span class='fl'>30</span>),
                     <span class='kw'>alt</span> <span class='kw'>=</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/c'>c</a></span>(<span class='st'>"ACT"</span>,<span class='st'>"ACGTCG"</span>,<span class='st'>"AGG"</span>,<span class='st'>"ACT"</span>,<span class='st'>"ACGTCG"</span>,<span class='st'>"AGG"</span>,<span class='st'>"ACT"</span>,<span class='st'>"ACGTCG"</span>,<span class='st'>"AGG"</span>),
                    <span class='kw'>ref</span> <span class='kw'>=</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/c'>c</a></span>(<span class='st'>"A"</span>,<span class='st'>"G"</span>,<span class='st'>"A"</span>,<span class='st'>"A"</span>,<span class='st'>"G"</span>,<span class='st'>"AGG"</span>,<span class='st'>"A"</span>,<span class='st'>"G"</span>,<span class='st'>"A"</span>),
                     <span class='kw'>ref_depth</span> <span class='kw'>=</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/c'>c</a></span>(<span class='fl'>12</span>,<span class='fl'>11</span>,<span class='fl'>9</span>,<span class='fl'>9</span>,<span class='fl'>12</span>,<span class='fl'>8</span>,<span class='fl'>13</span>,<span class='fl'>14</span>,<span class='fl'>8</span>),
                     <span class='kw'>alt_depth</span> <span class='kw'>=</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/c'>c</a></span>(<span class='fl'>2</span>,<span class='fl'>20</span>,<span class='fl'>10</span>,<span class='fl'>2</span>,<span class='fl'>10</span>,<span class='fl'>15</span>,<span class='fl'>20</span>,<span class='fl'>20</span>,<span class='fl'>100</span>),
                     <span class='kw'>SYMBOL</span> <span class='kw'>=</span> <span class='st'>"BCL2"</span>,
                     <span class='kw'>Consequence</span> <span class='kw'>=</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/rep'>rep</a></span>(<span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/c'>c</a></span>(<span class='st'>"exon1"</span>,<span class='st'>"exon1"</span>,<span class='st'>"exon2"</span>),<span class='kw'>times</span><span class='kw'>=</span><span class='fl'>3</span>),
                     <span class='kw'>qual</span> <span class='kw'>=</span> <span class='fl'>49</span>)

<span class='no'>clinicalData</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/data.frame'>data.frame</a></span>(<span class='kw'>SampleName</span> <span class='kw'>=</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/c'>c</a></span>(<span class='st'>"D1.Screen.Rel"</span>,<span class='st'>"D1.Cyc1.Rel"</span>,
                                          <span class='st'>"D1.Cyc2.Rel"</span>,<span class='st'>"D1.Cyc3.Rel"</span>),
                           <span class='kw'>PID</span> <span class='kw'>=</span> <span class='st'>"D1"</span>,
                           <span class='kw'>AgeDiagnosis</span> <span class='kw'>=</span> <span class='fl'>65</span>,
                            <span class='kw'>Time</span> <span class='kw'>=</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/c'>c</a></span>(<span class='st'>"Screen"</span>,<span class='st'>"Cyc1"</span>,<span class='st'>"Cyc2"</span>,<span class='st'>"Cyc3"</span>),
                          <span class='kw'>Sex</span> <span class='kw'>=</span> <span class='st'>"F"</span>,
                           <span class='kw'>BlastPerc</span> <span class='kw'>=</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/c'>c</a></span>(<span class='fl'>80</span>,<span class='fl'>5</span>,<span class='fl'>7</span>,<span class='fl'>40</span>))

<span class='no'>import_indels</span> <span class='kw'>&lt;-</span> <span class='fu'>import_any</span>(<span class='kw'>variants</span> <span class='kw'>=</span> <span class='no'>indels</span>,
                            <span class='kw'>patientID</span> <span class='kw'>=</span> <span class='st'>"D1"</span>,
                            <span class='kw'>studyGenes</span> <span class='kw'>=</span> <span class='st'>"BCL2"</span>,
                            <span class='kw'>minQual</span> <span class='kw'>=</span> <span class='fl'>20</span>,
                            <span class='kw'>tidy</span><span class='kw'>=</span><span class='fl'>TRUE</span>,
                            <span class='kw'>clinicalData</span> <span class='kw'>=</span> <span class='no'>clinicalData</span>)</div><div class='output co'>#&gt; IMPACT is missing and will be filled with NAs.</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      
      <li><a href="#details">Details</a></li>
      
      <li><a href="#examples">Examples</a></li>
    </ul>

  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by Anna Quaglieri, Christoffer Flensburg.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.9000.</p>
</div>
      </footer>
   </div>

  

  </body>
</html>

